---
title: "Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs"
author: "Joanna Yeung"
date: '2025-01-27'
output: html_document
---
```{r}
library(readr)
library(GenomicFeatures)
library(DESeq2)
library(apeglm)
library(gridExtra)
library(ggplot2)
library(pheatmap)
library(msigdbr)
library(stringr)
library(org.Hs.eg.db)
library(ggrepel)
library(rtracklayer)
library(cowplot)
library(readxl)
library(EnsDb.Hsapiens.v79)
library(viridis)
library(GenomicRanges)
library(rtracklayer)
library(RColorBrewer)
library(cowplot)
library(preprocessCore)
library(scales)
library(ChIPseeker)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(org.Hs.eg.db)
library(BSgenome.Hsapiens.UCSC.hg38)
library(MotifDb)
library(motifmatchr)
library(Biostrings)
library(clusterProfiler)
library(pheatmap)
library(dplyr)
```

# annotation colors for pheatmap
```{r}
hm_cols_RNA <- list(Timepoint=c("0"="#440154FF", "3"="#3B528BFF", "10"= "#4DA6A3", "14"="#21908CFF", "21"="#5DC863FF", "28"="#FDE725FF"), 
                Treatment=c("Cycling"="#F8766D", "Doxo"="#B3C543", "Palbo"="#E0ADCD", "DoxovsPalbo"="#CCEBC5"), 
                Biorep=c("1"="orange", "2"="light blue", "3"="pink"), 
                log2FC=c("vs Cyc"="#FBB4AE","vs day3"="#B3CDE3", "DoxovsPalbo"="#CCEBC5"), 
                clusters_cutree8=brewer.pal(8, "Set2"), 
                Clusters=c("5"="#66C2A5","2"="#A6D854","1"="#FC8D62", "7"="#8DA0CB","4"="#E78AC3","6"="#E78AC3","3"="#FFD92F","8"="#E5C494"), 
                SASP = c("Yes"="dark green", "No"="white"),
                Cell_Cycle = c("Yes"="dark blue", "No"="white"), 
                E2F = c("Yes"="dark blue", "No"="white"), 
                p53_All=c("Yes"="#00B0F0", "No"="white"),
                NFKB_All=c("Yes"="red", "No"="white"))
names(hm_cols_RNA$clusters_cutree8) <- as.factor(1:8)


hm_cols_ATAC <- list(Timepoint=c("0"="#440154FF", "3"="#3B528BFF", "10"= "#4DA6A3", "14"="#21908CFF", "21"="#5DC863FF", "28"="#FDE725FF"), 
                Treatment=c("None"="#F8766D", "Doxo"="#B3C543", "Palbo"="#E0ADCD", "PvsD"="#CCEBC5"),
                Biorep=c("1"="orange", "2"="light blue", "3"="pink", "4"="light green"), 
                log2FC=c("vs Cyc"="#FBB4AE","vs day3"="#B3CDE3", "PvsD"="#CCEBC5"),
                Peak_anno=c("Distal"="#B15A28", "Gene Body"="#7F54A2", "Promoter"="#A7CEE2"), 
                Doxo=c("Yes"="#B3C543", "No"="white"),
                Palbo=c("Yes"="#E0ADCD", "No"="white")
                )
```

# load in workspace with DESeq results and rlog counts matrix of genes. 
```{r}
load("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Global_transcriptome_and_chromatin_accessibility_is_dynamic_in_both_types_of_therapy_induced_senescence/workspace.RData")
```

#### universe (background gene set) for GO enrichment analysis is all genes expressed
```{r}
universe <- genes_anno_type[match(gsub('\\.[0-9]*$', '', row.names(acrossGroups)), genes_anno_type$ensembl_id), ]
```

# GO enrichment analysis on k-means clusters from sigMat (significantly changing genes across all conditions, based on Likelihood ratio test)

## *function to get GO enriched terms*:
## arguments:
### gene= vector of genes to test for GO enrichment (gene ensembl format)
### universe=background gene set 
### Database=database containing groups of genes to GO annotation term e.g. msigdb GO:BP
```{r}
getGOenrichment <-function(gene,universe,Database){
  #gene_entrezID <-  AnnotationDbi::select(EnsDb.Hsapiens.v79, gene[!is.na(gene)], "ENTREZID", "SYMBOL") #convert gene IDs from SYMBOL format to ENTREZ ID format (was using v79, new code is using v107)
  gene_entrezID <- genes_anno_type[match(gene, genes_anno_type$gene_id), ]
  gene_go <- enricher(gene_entrezID$entrezid, TERM2GENE=Database, universe=universe) #enrichment analysis with hypergeometric test
  gene_go <- setReadable(gene_go, OrgDb = org.Hs.eg.db, keyType="ENTREZID") #convert enrichment results from ENTREZ ID format to SYMBOL for easier interpretation
return(gene_go)
} 
```

# assign significant genes to relevant k-means clusters & drug + cell cycle arrest categories. 
```{r}
genes_in_clusters <- lapply(levels(clusterDF$Clusters), function(x){row.names(clusterDF[clusterDF$Clusters %in% x, ])})
genes_in_clusters[[1]] <- unlist(genes_in_clusters[1:2])
genes_in_clusters[[6]] <- unlist(genes_in_clusters[5:6])
genes_in_clusters <- genes_in_clusters[c(1,3,4,6,7,8)]
cluster_names <- c("CellCycle", "Palbo1", "Palbo2", "Doxo1", "Doxo2", "Shared") 
names(genes_in_clusters) <- cluster_names
```

# RNA-seq: GO enrichment on significant genes
```{r}
# GO enrichment analysis with msigdb GO:BP database:
GOBP_enrichment_clusters <- list()
# GO enrichment analysis with msigdb REACT database:
GO_enrichment_clusters_REACT <- list()
# GO enrichment analysis with msigdb Hallmark database:
GO_enrichment_clusters_HallMark <- list()
# GO enrichment analysis with msigdb REACT database:
GO_enrichment_clusters_REACT <- list()

# GO enrichment analysis with msigdb GO:MF database:
GO_enrichment_clusters_MF <- list()

for(i in 1:length(genes_in_clusters)){
  
GOBP_enrichment_clusters[[i]] <- getGOenrichment(genes_in_clusters[[i]], as.character(universe$entrezid),msigdbr_t2g_GO_BP) # do GO enrichment analysis with background gene set set as all gene expressed, database used is msigdb GO:BP

GO_enrichment_clusters_REACT[[i]] <- getGOenrichment(genes_in_clusters[[i]], as.character(universe$entrezid),msigdbr_t2g_REACT)

GO_enrichment_clusters_HallMark[[i]] <- getGOenrichment(genes_in_clusters[[i]], as.character(universe$entrezid),msigdbr_t2g_H)

GO_enrichment_clusters_REACT[[i]] <- getGOenrichment(genes_in_clusters[[i]], as.character(universe$entrezid),msigdbr_t2g_REACT)

 GO_enrichment_clusters_MF[[i]] <- getGOenrichment(genes_in_clusters[[i]], as.character(universe$entrezid),msigdbr_t2g_GO_MF)
}
```

### RNA-seq: format GO enrichment terms from each cluster into matrix of -log10(padj values)
```{r}
# convert each clusters' GO enrichment terms into dataframes
GO_enrichment_clusters_REACT_df <- list()
 GO_enrichment_clusters_REACT_df <- list()
 GO_enrichment_clusters_HallMark_df <- list()
 GOBP_enrichment_clusters_df <- list()
for(i in 1:length(GO_enrichment_clusters_REACT)){
  GO_enrichment_clusters_REACT_df[[i]] <- as.data.frame(GO_enrichment_clusters_REACT[[i]])
   GO_enrichment_clusters_REACT_df[[i]] <- as.data.frame(GO_enrichment_clusters_REACT[[i]])
 GO_enrichment_clusters_HallMark_df[[i]] <- as.data.frame(GO_enrichment_clusters_HallMark[[i]])
 GOBP_enrichment_clusters_df[[i]] <- as.data.frame(GOBP_enrichment_clusters[[i]])
}

 # indicate names of each list as the cluster number
 names(GO_enrichment_clusters_REACT_df) <- cluster_names 
 names(GO_enrichment_clusters_REACT_df) <- cluster_names 
 names(GO_enrichment_clusters_HallMark_df) <- cluster_names 
 names(GOBP_enrichment_clusters_df) <- cluster_names 
 
 # remove any cluster of genes that have 0 GO enrichment terms
 GO_enrichment_clusters_REACT_df <-  GO_enrichment_clusters_REACT_df[sapply( GO_enrichment_clusters_REACT_df, nrow)>0]
 GO_enrichment_clusters_REACT_df <-  GO_enrichment_clusters_REACT_df[sapply( GO_enrichment_clusters_REACT_df, nrow)>0]
 GO_enrichment_clusters_HallMark_df <-  GO_enrichment_clusters_HallMark_df[sapply( GO_enrichment_clusters_HallMark_df, nrow)>0]
 GOBP_enrichment_clusters_df <-  GOBP_enrichment_clusters_df[sapply( GOBP_enrichment_clusters_df, nrow)>0]
 
 # indicate which cluster and database each dataframe comes from 
 for(i in 1:length(GO_enrichment_clusters_REACT_df)){
  GO_enrichment_clusters_REACT_df[[i]]$Cluster <- names(GO_enrichment_clusters_REACT_df)[i]
  GO_enrichment_clusters_REACT_df[[i]]$Database <- "REACT"
 }
  for(i in 1:length(GO_enrichment_clusters_REACT_df)){
   GO_enrichment_clusters_REACT_df[[i]]$Cluster <-  names(GO_enrichment_clusters_REACT_df)[i]
   GO_enrichment_clusters_REACT_df[[i]]$Database <- "REACTOME"
  }
  for(i in 1:length(GO_enrichment_clusters_HallMark_df)){
 GO_enrichment_clusters_HallMark_df[[i]]$Cluster <- names(GO_enrichment_clusters_HallMark_df)[i]
 GO_enrichment_clusters_HallMark_df[[i]]$Database <- "Hallmark"
  }
 
  for(i in 1:length(GOBP_enrichment_clusters_df)){
 GOBP_enrichment_clusters_df[[i]]$Cluster <- names(GOBP_enrichment_clusters_df)[i]
  GOBP_enrichment_clusters_df[[i]]$Database <- "GO:BP"
  }

# combine GO enrichment results from different databases into 1 dataframe for each cluster of genes. 
GO_results_list <- vector()
GO_results_list$CellCycle <- rbind(GO_enrichment_clusters_REACT_df$CellCycle, GO_enrichment_clusters_HallMark_df$CellCycle, GO_enrichment_clusters_REACT_df$CellCycle, GOBP_enrichment_clusters_df$CellCycle)

GO_results_list$Palbo1 <- rbind(GO_enrichment_clusters_REACT_df$Palbo1, GO_enrichment_clusters_HallMark_df$Palbo1 , GO_enrichment_clusters_REACT_df$Palbo1, GOBP_enrichment_clusters_df$Palbo1)

GO_results_list$Palbo2 <- rbind(GO_enrichment_clusters_REACT_df$Palbo2, GO_enrichment_clusters_HallMark_df$Palbo2, GO_enrichment_clusters_REACT_df$Palbo2, GOBP_enrichment_clusters_df$Palbo2)

GO_results_list$Doxo1 <- rbind(GO_enrichment_clusters_REACT_df$Doxo1, GO_enrichment_clusters_HallMark_df$Doxo1, GO_enrichment_clusters_REACT_df$Doxo1, GOBP_enrichment_clusters_df$Doxo1)

GO_results_list$Doxo2 <- rbind(GO_enrichment_clusters_REACT_df$Doxo2, GO_enrichment_clusters_HallMark_df$Doxo2, GO_enrichment_clusters_REACT_df$Doxo2, GOBP_enrichment_clusters_df$Doxo2)

GO_results_list$Shared <- rbind(GO_enrichment_clusters_REACT_df$Shared, GO_enrichment_clusters_HallMark_df$Shared, GO_enrichment_clusters_REACT_df$Shared, GOBP_enrichment_clusters_df$Shared)

# order GO enrichment results from different databases by padj values
for(i in 1:length(GO_results_list)){
  GO_results_list[[i]] <- GO_results_list[[i]][order(GO_results_list[[i]]$p.adjust), ]
}

# combine dataframes of each cluster of gene's GO enrichment results into 1 master dataframe
GO_results_df <- do.call(rbind, GO_results_list)

# create matrix of -log10pvalues of the top 15 enriched pathways for each cluster

GO_results_list_mat <- matrix(nrow=length(unique(unlist(lapply(GO_results_list, function(GO_results_list){GO_results_list$ID[1:50]})))), ncol=length(GO_results_list)) # make empty matrix, number of columns are the clusters of genes and number of total top GO terms of all clusters

row.names(GO_results_list_mat) <- unique(unlist(lapply(GO_results_list, function(GO_results_list){GO_results_list$ID[1:50]}))) # row names are top 12 GO enrichment terms for each cluster combined
colnames(GO_results_list_mat) <- cluster_names  # column names are clusters

# pathways that are not enriched in the cluster are listed as NA so just change the value to 0 so that it can be plotted in heatmap
for(i in 1:ncol(GO_results_list_mat)){
GO_results_list_mat[ ,i] <- -log10(GO_results_list[[i]][match(row.names(GO_results_list_mat), GO_results_list[[i]]$ID), ]$p.adjust)
GO_results_list_mat[ ,i] <- ifelse(GO_results_list_mat[ ,i] %in% NA, 0, GO_results_list_mat[ ,i])
}
GO_results_list_mat <- GO_results_list_mat[!is.na(row.names(GO_results_list_mat)), ] # remove padj values that are NA 

GO_anno_colors <-  list(database=c("GO:BP"="#3B4992FF", "Hallmark"="#EE0000FF", "REACT"="#008B45FF", "REACTOME" ="#631879FF"), Cluster=c("CellCycle"="#66C2A5", "Palbo1"="#FC8D62", "Palbo2"="#8DA0CB","Doxo1"="#E78AC3","Doxo2"="#FFD92F", "Shared"="#E5C494"))
names(GO_anno_colors$Cluster) <- cluster_names 
# annotation dataframe for rows of heatmap to indicate which database GO term came from 
GO_anno <- data.frame(row.names=row.names(GO_results_list_mat), database=GO_results_df[match(row.names(GO_results_list_mat), GO_results_df$ID), ]$Database)
# annotation dataframe for columns of heatmap to color by Cluster colors. 
Cluster_anno <- data.frame(Cluster=cluster_names , row.names = colnames(GO_results_list_mat))
```

# RNA-seq: GSEA Analysis
```{r}
ranked_res_forGSEA <- list()
for(i in 1:length(LFCResultsall_RNA)){
  
  rank <- LFCResultsall_RNA[[i]]
  
  # Assign a default p-value for genes with NA padj (maximum p-value)
  rank$pvalue[is.na(rank$padj)] <- 1 
  
  # Create a ranked list based on log2FoldChange and -log10(p-value)
  rank$rank_metric <- sign(rank$log2FoldChange) * -log10(rank$pvalue)
  
  # Sort by the ranking metric
  rank <- rank[order(rank$rank_metric, decreasing = TRUE), ]
  
  # Store the ranked values
  ranked_res_forGSEA[[i]] <- rank$rank_metric
  names(ranked_res_forGSEA[[i]]) <- row.names(rank)

  # Convert gene IDs to Entrez IDs if available
  entrez_ids <- genes_anno_type[match(names(ranked_res_forGSEA[[i]]), genes_anno_type$ensembl_id), ]$entrezid
  names(ranked_res_forGSEA[[i]]) <- ifelse(is.na(entrez_ids), names(ranked_res_forGSEA[[i]]), entrez_ids)
}

names(ranked_res_forGSEA) <- names(LFCResultsall_RNA)

# run GSEA analysis using: 
GSEA_Results_H <- list() # msigdb Hallmark database
GSEA_Results_REACT <- list() # msigdb KEGG database
GSEA_Results_REACT <- list() # msigdb Reactome database
GSEA_plots_H <- list()
GSEA_plots_REACT <- list()
GSEA_plots_REACT <- list()
for(i in 1:length(ranked_res_forGSEA)){
 GSEA_Results_H[[i]] <- GSEA(ranked_res_forGSEA[[i]], TERM2GENE = msigdbr_t2g_H) 
 GSEA_Results_REACT[[i]] <- GSEA(ranked_res_forGSEA[[i]], TERM2GENE = msigdbr_t2g_REACT) 
 GSEA_Results_REACT[[i]] <- GSEA(ranked_res_forGSEA[[i]], TERM2GENE = msigdbr_t2g_REACT) 
}

names(GSEA_Results_H) <- names(LFCResultsall_RNA)
names(GSEA_Results_REACT) <- names(LFCResultsall_RNA)
names(GSEA_Results_REACT) <- names(LFCResultsall_RNA)

# generate running score plots for GSEA results
# GSEA Hallmark
for(i in 1:length(GSEA_Results_H)){
  maxplots <- length(GSEA_Results_H[[i]]$ID)
GSEA_plots_H[[i]] <- list()
GSEA_plots_KEGG[[i]] <- list()
GSEA_plots_REACT[[i]] <- list()
  for(j in 1:maxplots){
  GSEA_plots_H[[i]][[j]] <- gseaplot(GSEA_Results_H[[i]], geneSetID = j, by = "runningScore", title = paste0(names(GSEA_Results_H)[i], ": " , GSEA_Results_H[[i]]$ID[j]) )
  }
  names(GSEA_plots_H[[i]]) <- GSEA_Results_H[[i]]$ID[1:maxplots]
}

# GSEA KEGG
for (i in 1:length(GSEA_Results_KEGG)) {
  # Check if the current object is not empty
  if (!is.null(GSEA_Results_KEGG[[i]]) && length(GSEA_Results_KEGG[[i]]$ID) > 0) {
    maxplots <- length(GSEA_Results_KEGG[[i]]$ID)
    GSEA_plots_KEGG[[i]] <- list()  # Initialize a list to store plots for this object
    
    for (j in 1:maxplots) {
      GSEA_plots_KEGG[[i]][[j]] <- gseaplot(
        GSEA_Results_KEGG[[i]], 
        geneSetID = j, 
        by = "runningScore", 
        title = GSEA_Results_KEGG[[i]]$ID[j]
      )
    }
    # Assign names to the plots
    names(GSEA_plots_KEGG[[i]]) <- GSEA_Results_KEGG[[i]]$ID[1:maxplots]
  } else {
    # If empty, set the current slot in GSEA_plots_KEGG to NULL
    GSEA_plots_KEGG[[i]] <- NULL
  }
}

# GSEA Reactome
for (i in 1:length(GSEA_Results_REACT)) {
  # Check if the current object is not empty
  if (!is.null(GSEA_Results_REACT[[i]]) && length(GSEA_Results_REACT[[i]]$ID) > 0) {
    maxplots <- length(GSEA_Results_REACT[[i]]$ID)
    GSEA_plots_REACT[[i]] <- list()  # Initialize a list to store plots for this object
    
    for (j in 1:maxplots) {
      GSEA_plots_REACT[[i]][[j]] <- gseaplot(
        GSEA_Results_REACT[[i]], 
        geneSetID = j, 
        by = "runningScore", 
        title = GSEA_Results_REACT[[i]]$ID[j]
      )
    }
    # Assign names to the plots
    names(GSEA_plots_REACT[[i]]) <- GSEA_Results_REACT[[i]]$ID[1:maxplots]
  } else {
    # If empty, set the current slot in GSEA_plots_REACT to NULL
    GSEA_plots_REACT[[i]] <- NULL
  }
}

# assign names to plots based on which DESeq comparison GSEA was done on. 
names(GSEA_plots_H) <- names(LFCResultsall_RNA)
names(GSEA_plots_KEGG) <- names(which(unlist(lapply(GSEA_Results_KEGG, function(x){length(x$ID)})) >0))
names(GSEA_plots_REACT) <- names(which(unlist(lapply(GSEA_Results_REACT, function(x){length(x$ID)})) >0))
```

# RNA-seq: import IPA upstream regulators & pathways
```{r}
setwd("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/IPA_files/output/")
IPA_upstream_regs <- list()
IPA_pathways <- list()
# read in information for enriched IPA pathways per hierarchical cluster (clustering based on scaled rlog normalized counts)
for(i in 1:length(cluster_names)){
IPA_pathways[[i]]<- read_tsv(paste0(cluster_names[i], "_IPA_Pathways.txt"), skip=2)
# read in information for enriched upstream regulators 
IPA_upstream_regs[[i]] <- read.table(paste0(cluster_names[i], "_IPA_UpstreamRegs.txt"), sep="\t", skip=2, header=T)
IPA_upstream_regs[[i]] <- IPA_upstream_regs[[i]][!grepl("mir|miR|Mir", IPA_upstream_regs[[i]]$Upstream.Regulator), ] # remove upstream regulators that are miRNAs (uncertain if they will be informative & there is a lot of them)
}

#extract ids of all expressed genes
expressedGenes <- LFCResultsall_RNA[[1]]$SYMBOL

# keep only upstream regulators that are part of expressed genes or have lower case letters (these are upstream regulators that could be miRNAs or protein complexes)
IPA_upstream_regs <- lapply(IPA_upstream_regs, function(IPA_upstream_regs){IPA_upstream_regs[c(grep('[:lower:]', IPA_upstream_regs$Upstream.Regulator), which(IPA_upstream_regs$Upstream.Regulator %in% expressedGenes)), ]})

names(IPA_upstream_regs) <- cluster_names

names(IPA_pathways) <- cluster_names
```

## plotting padj values of top IPA predicted upstream regulators of all clusters
```{r}
create_ipa_regs_matrix <- function(IPA_upstream_regs, cluster_names, filter_keyword = "complex", top_n = 25) {
  # Extract unique upstream regulators based on the filter keyword
  unique_upstream_regs <- unique(unlist(lapply(IPA_upstream_regs, function(x) {
    x$Upstream.Regulator[grepl(filter_keyword, x$Molecule.Type)][1:top_n]
  })))

  # Create a matrix to store -log10 p-values
  IPA_complex_regs_mat <- matrix(
    nrow = length(unique_upstream_regs),
    ncol = length(IPA_upstream_regs)
  )
  row.names(IPA_complex_regs_mat) <- unique_upstream_regs # Set row names to upstream regulators
  colnames(IPA_complex_regs_mat) <- cluster_names        # Set column names to cluster names

  # Fill in the matrix with -log10 p-values
  for (i in seq_len(ncol(IPA_complex_regs_mat))) {
    # Match and extract p-values for each cluster
    IPA_complex_regs_mat[, i] <- -log10(IPA_upstream_regs[[i]][
      match(row.names(IPA_complex_regs_mat), IPA_upstream_regs[[i]]$Upstream.Regulator),
      "p.value.of.overlap"
    ])

    # Replace NA values with 0 for heatmap compatibility
    IPA_complex_regs_mat[, i][is.na(IPA_complex_regs_mat[, i])] <- 0
  }

  return(IPA_complex_regs_mat)
}

IPA_group_regs_mat <- create_ipa_regs_matrix(
  IPA_upstream_regs = IPA_upstream_regs,
  cluster_names = cluster_names,
  filter_keyword = "group",
  top_n = 25
)

IPA_TF_regs_mat <- create_ipa_regs_matrix(
  IPA_upstream_regs = IPA_upstream_regs,
  cluster_names = cluster_names,
  filter_keyword = "transcription",
  top_n = 25
)

# plot heatmap of -log10 pvalue of overlap of enriched complex regulators
pheatmap(IPA_group_regs_mat, cluster_cols = F, colorRampPalette(brewer.pal(n = 7, name ="Greens"))(100), breaks = seq(0, 10,length.out = 100))

pheatmap(IPA_TF_regs_mat, cluster_cols = F, colorRampPalette(brewer.pal(n = 7, name ="Greens"))(100), breaks = seq(0, 10,length.out = 100))
```

# RNA-seq: make matrix with log2FC values of significant genes
```{r}
# get log2FC values of significantly changing genes over time
sig_LFCMat <- matrix(ncol=length(LFCResultsall_RNA), nrow=nrow(clusterDF))
colnames(sig_LFCMat) <- names(LFCResultsall_RNA)
row.names(sig_LFCMat) <-row.names(LFCResultsall_RNA[[10]][row.names(LFCResultsall_RNA[[10]]) %in% sigGenes, ])
for(i in 1:ncol(sig_LFCMat)){
sig_LFCMat[ ,i] <- LFCResultsall_RNA[[i]][row.names(LFCResultsall_RNA[[i]]) %in% sigGenes, ]$log2FoldChange
}

row.names(sig_LFCMat) <-  ifelse(genes_anno_type[match(row.names(sig_LFCMat), genes_anno_type$ensembl_id), ]$gene_id %in% NA, row.names(sig_LFCMat), genes_anno_type[match(row.names(sig_LFCMat), genes_anno_type$ensembl_id), ]$gene_id)

row.names(ordered_sigMat)[grep("^H1.[0-9]", row.names(ordered_sigMat))]  <- gsub("*H1.", "H1-",row.names(ordered_sigMat[grepl("^H1.[0-9]", row.names(ordered_sigMat)), ]) )

sig_LFCMat <- sig_LFCMat[match(row.names(ordered_sigMat), row.names(sig_LFCMat)), ] # order based on k-means & hierarchical ordering of rlog counts. 

# get log2FC values of all expressed genes over time 
LFCMat <- matrix(ncol=length(LFCResultsall_RNA), nrow=nrow(LFCResultsall_RNA[[1]]))
colnames(LFCMat) <- names(LFCResultsall_RNA)
row.names(LFCMat) <-row.names(LFCResultsall_RNA[[1]])
for(i in 1:ncol(LFCMat)){
LFCMat[ ,i] <- LFCResultsall_RNA[[i]]$log2FoldChange
}

row.names(LFCMat) <-  ifelse(genes_anno_type[match(row.names(LFCMat), genes_anno_type$ensembl_id), ]$gene_id %in% NA, row.names(LFCMat), genes_anno_type[match(row.names(LFCMat), genes_anno_type$ensembl_id), ]$gene_id)
```

# RNA-seq: cell cycle arrest signatures
```{r}
# extract genes under Hallmark terms P53 PATHWAY & TNFA SIGNALING VIA NFKB
p53_Hallmark <- msigdbr_df_H[msigdbr_df_H$gs_name %in% "HALLMARK_P53_PATHWAY", ]$gene_symbol
NFKB_Hallmark <- msigdbr_df_H[msigdbr_df_H$gs_name %in% "HALLMARK_TNFA_SIGNALING_VIA_NFKB", ]$gene_symbol

# filter for E2F target genes from IPA & Hallmark E2F Targets pathway term. 
E2F_Targets=unique(c(unlist(lapply(IPA_upstream_regs, function(x){str_split(x[grep("E2F", x$Upstream.Regulator), ]$Target.Molecules.in.Dataset, ",")})), 
              msigdbr_df_H[msigdbr_df_H$gs_name %in% "HALLMARK_E2F_TARGETS", ]$gene_symbol
))

# make log2FC matrix of cell cycle genes (based on KEGG Cell Cycle pathway term) going down with drug treatment
CellCycle_sig_LFCMat <- sig_LFCMat[row.names(sig_LFCMat) %in% msigdbr_df_KEGG[msigdbr_df_KEGG$gs_name %in% "KEGG_CELL_CYCLE", ]$gene_symbol & row.names(sig_LFCMat) %in% row.names(clusterDF[clusterDF$Clusters %in% c("5", "2"), ]), ]

anno_sigLFC <- data.frame(
  row.names = colnames(sig_LFCMat),
  log2FC = c(rep("vs Cyc", 4), rep("vs day3", 3), rep("vs Cyc", 5), rep("vs day3", 4), rep("DoxovsPalbo",4)),
  Treatment = c(rep("Doxo", 7), rep("Palbo", 9), rep("DoxovsPalbo",4)),
  Timepoint = c("3", "10", "14", "21", "10", "14", "21", "3", "10", "14", "21", "28", "10", "14", "21", "28","3", "10", "14", "21")
)

# add annotation to significant genes based on if they are E2F targets. 
clusterDF$E2F <- ifelse(row.names(clusterDF) %in% E2F_Targets, "Yes", "No") # annotate sig genes that are E2F targets

row.names(clusterDF)[grep("^H1.[0-9]", row.names(clusterDF))]  <- gsub("*H1.", "H1-",row.names(clusterDF[grepl("^H1.[0-9]", row.names(clusterDF)), ]) ) # correction: to match with row.names in sig_LFCMat, replace - with . for linker histone genes. 
```

# ATAC-seq: plot Tn5 insertion sites around E2F1 motifs overlapping down peaks to show cell cycle signature is also reflected at the chromatin accessibility level in both drugs. 
```{r}
# filter for GRanges of up peaks in each drug and cell cycle arrest category 
Down_DrugandArrestCat <- list(
  PalboDownvsCyc=allDownvsCycling$Palbo[!allDownvsCycling$Palbo %over% allDownvsCycling$Doxo, ], 
  sharedDownvsCyc=allDownvsCycling$Palbo[allDownvsCycling$Palbo %over% allDownvsCycling$Doxo, ],
  DoxoDownvsCyc=allDownvsCycling$Doxo[!allDownvsCycling$Doxo %over% allDownvsCycling$Palbo, ], 
  PalboDownvsday3=allDownvsday3$Palbo[!allDownvsday3$Palbo %over% allDownvsday3$Doxo, ], 
  sharedDownvsday3=allDownvsday3$Palbo[allDownvsday3$Palbo %over% allDownvsday3$Doxo, ], 
  DoxoDownvsday3=allDownvsday3$Doxo[!allDownvsday3$Doxo %over% allDownvsday3$Palbo, ])

# filter for GRanges of Down peaks in each drug and cell cycle arrest category but without specifying shared Down peaks. 
Down_DrugandArrest_PvsD <- list(
  PalboDownvsCyc=allDownvsCycling$Palbo, 
  DoxoDownvsCyc=allDownvsCycling$Doxo, 
  PalboDownvsday3=allDownvsday3$Palbo,
  DoxoDownvsday3=allDownvsday3$Doxo)

# load in HOCOMOCO motif sequence database
load("/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/post_peakcalling_analysis/motifs/HOCOMOCO_v11_matrix.RData")
# scan for E2F1 motifs
E2F1_HMv11 <- matchPWM(HOCOMOCO_v11_matrix$E2F1_HUMAN.H11MO.0.A, BSgenome.Hsapiens.UCSC.hg38, with.score=T)

# filter for E2F1 motifs that overlap with Down peaks
E2F1_ATAC_Down_DrugandArrestCat <- lapply(Down_DrugandArrestCat, function(x){E2F1_HMv11[E2F1_HMv11 %over% x]})
E2F1_ATAC_Down_DrugandArrest_PvsD <- lapply(Down_DrugandArrest_PvsD, function(x){E2F1_HMv11[E2F1_HMv11 %over% x]})

directory <- "/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/deepTools_files/input_beds"
# Export each GRanges object as a BED file
for(i in 1:length(E2F1_ATAC_Down_DrugandArrestCat)){
export.bed(E2F1_ATAC_Down_DrugandArrestCat[[i]], con = paste0(directory, "/", names(E2F1_ATAC_Down_DrugandArrestCat)[i], "_E2F1_HMv11.bed"))
}

for(i in 1:length(E2F1_ATAC_Down_DrugandArrest_PvsD)){
export.bed(E2F1_ATAC_Down_DrugandArrest_PvsD[[i]], con = paste0(directory, "/", names(E2F1_ATAC_Down_DrugandArrest_PvsD)[i], "_PvsD_E2F1_HMv11.bed"))
}

# generate input files for deepTools scripts
# directory paths to bed files 
writeLines(dir(directory, full.names=T, pattern=".bed"), con=paste0(directory, "/regions.txt"))

# filenames to be assigned to plots. 
filenames <- gsub(".bed", "", dir(directory, pattern=".bed"))
writeLines(filenames, con=paste0(directory, "/filename.txt"))
```

# submit bash job to plot Tn5 cut sites around E2F1 motifs in down peaks
```{bash}
mergedcuts="/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/Cyclingmerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day3_Doxomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day10_Doxomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day14_Doxomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day21_Doxomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day3_Palbomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day10_Palbomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day14_Palbomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day21_Palbomerged_scaleFactornorm_cutsCoverage.bw /lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day28_Palbomerged_scaleFactornorm_cutsCoverage.bw"

samplenames="Cycling day3_Doxo day10_Doxo day14_Doxo day21_Doxo day3_Palbo day10Palbo day14_Palbo day21_Palbo day28_Palbo"

cd /lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/deepTools_files/output_plots
sbatch -p risc,hpc --array=1-10 \
--export=bigwigs="$mergedcuts",\
regions=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/deepTools_files/input_beds/regions.txt,\
samplenames="$samplenames",\
profile_outdir=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/deepTools_files/output_plots/plotProfile,\
heatmap_outdir=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/deepTools_files/output_plots/plotHeatmap,\
filename=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/deepTools_files/input_beds/filename.txt \
/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/deepTools_files/deeptools_Coverage_1kb.slurm
```

# ATAC-seq: import motif enrichment results under down peaks across all timepoint comparisons & drug treatment. 
```{r}
# load in process_motif_data function. 
load("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Supplemental_Figure_peaks_50kb_of_RNAclusters_heatmap_motif_enrichment/Functions/process_motif_data.RData")
```

```{r}
annoDownPeaks_AME_df <- process_motif_data("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Supplemental_Figure_Summary_Motif_enrichment_results_all_sig_peaks/motif_analysis_output/DESeq/annoDown", expressed_genes = expressedGenes)

annoDownPeaks_AME_df$Comparison <- gsub("\\..*", "", row.names(annoDownPeaks_AME_df))
annoDownPeaks_AME_df$Comparison <- factor(annoDownPeaks_AME_df$Comparison, levels=unique(annoDownPeaks_AME_df$Comparison)[c(15,1,5,9,2,6,10,16,3,7,11,13,4,8,12,14)]) # column to identify DESeq comparison

annoDownPeaks_AME_df$Group <- ifelse(grepl("DoxovsCycling", annoDownPeaks_AME_df$Comparison), "DoxovsCycling", ifelse(grepl("PalbovsCycling", annoDownPeaks_AME_df$Comparison), "PalbovsCycling", ifelse(grepl("Doxovsday3Doxo", annoDownPeaks_AME_df$Comparison), "Doxovsday3Doxo", "Palbovsday3Palbo"))) # column to identify relative to cycling or day 3 comparison and drug treatment. 

annoDownPeaks_AME_df$Timepoint <-  gsub("(day[0-9]{1,2}).*", "\\1", annoDownPeaks_AME_df$Comparison) # column to identify which timepoint is compared to Cycling or day 3 controls, by removing everything after day[0-9] where [0-9] is 1 or 2 digits in the Comparison column. 
annoDownPeaks_AME_df$Timepoint <- factor(annoDownPeaks_AME_df$Timepoint, levels=c("day3", "day10", "day14", "day21", "day28"))
```

# ATAC-seq: get peak ids for peaks that are true positive peaks for E2F motif from motif enrichment analysis (MEME-AME)
```{r}
input_dir <- "/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Supplemental_Figure_Summary_Motif_enrichment_results_all_sig_peaks/motif_analysis_output/DESeq/annoDown"

out_dir <- "/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/E2F_tp_peaks"

# generate input file containing directory paths to sequences.tsv file from MEME-AME output. 
writeLines(paste0(dir(input_dir, full.names = T), "/sequences.tsv"), con=paste0(out_dir, "/DESeq_annoDown_sequencefiles.txt"))
```

```{bash}
# extract peak ids of true positive peaks from sequences.tsv file from MEME-AME output
cd /lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/E2F_tp_peaks

# for all E2F motifs
sbatch -p risc --array=1-16 \
--export=outdir=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/E2F_tp_peaks/all_E2F_motifs,\
inputfiles=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/E2F_tp_peaks/DESeq_annoDown_sequencefiles.txt,\
names=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Supplemental_Figure_Summary_Motif_enrichment_results_all_sig_peaks/motif_analysis_input/DESeq/annoDown/folderlist_annoDown_DESeq_peaks.txt,\
motif=E2F \
get_tp_peaks_motifs.slurm

rm $(ls|egrep '.err|.out')

# for E2F1 motif
sbatch -p risc --array=1-16 \
--export=outdir=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/E2F_tp_peaks/E2F1_motif,\
inputfiles=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/E2F_tp_peaks/DESeq_annoDown_sequencefiles.txt,\
names=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Supplemental_Figure_Summary_Motif_enrichment_results_all_sig_peaks/motif_analysis_input/DESeq/annoDown/folderlist_annoDown_DESeq_peaks.txt,\
motif=E2F1 \
get_tp_peaks_motifs.slurm

rm $(ls|egrep '.err|.out')
```

```{r}
# import peak ids of true positive down peaks containing any E2F motif
setwd("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/E2F_tp_peaks/all_E2F_motifs")

file_ids <- dir()
all_E2F_tp_peaks <- list()

for (i in seq_along(file_ids)) {
  file_path <- file_ids[i]  # Get full file name

  # Check if file is empty
  if (file.info(file_path)$size > 0) {
    all_E2F_tp_peaks[[file_path]] <- read_tsv(file_path, col_names = FALSE) %>% pull(X1)
  }
}

names(all_E2F_tp_peaks) <- gsub("_E2F_all_tp_Downpeaks_ids.tsv", "", names(all_E2F_tp_peaks))
```

```{r}
# import peak ids of true positive down peaks containing E2F1 motif
setwd("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/E2F_tp_peaks/E2F1_motif")

file_ids <- dir()
E2F1_tp_peaks <- list()

for (i in seq_along(file_ids)) {
  file_path <- file_ids[i]  # Get full file name

  # Check if file is empty
  if (file.info(file_path)$size > 0) {
    E2F1_tp_peaks[[file_path]] <- read_tsv(file_path, col_names = FALSE) %>% pull(X1)
  }
}

names(E2F1_tp_peaks) <- gsub("_E2F_all_tp_Downpeaks_ids.tsv", "", names(E2F1_tp_peaks))
```
# *FIGURES*

# RNA-seq: save GSEA running score for E2F targets for all timepoints relative to Cycling
```{r}
pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/GSEA_Hallmark_E2F_TARGETS_pathway.pdf", width=3, height=2.5)
GSEA_plots_H$`Doxo3 vs Cycling0`$`E2F TARGETS`+theme_classic()+ylim(-0.9,0.05)
GSEA_plots_H$`Doxo10 vs Cycling0`$`E2F TARGETS`+theme_classic()+ylim(-0.9,0.05)
GSEA_plots_H$`Doxo14 vs Cycling0`$`E2F TARGETS`+theme_classic()+ylim(-0.9,0.05)
GSEA_plots_H$`Doxo21 vs Cycling0`$`E2F TARGETS`+theme_classic()+ylim(-0.9,0.05)
GSEA_plots_H$`Palbo3 vs Cycling0`$`E2F TARGETS`+theme_classic()+ylim(-0.9,0.05)
GSEA_plots_H$`Palbo10 vs Cycling0`$`E2F TARGETS`+theme_classic()+ylim(-0.9,0.05)
GSEA_plots_H$`Palbo14 vs Cycling0`$`E2F TARGETS`+theme_classic()+ylim(-0.9,0.05)
GSEA_plots_H$`Palbo21 vs Cycling0`$`E2F TARGETS`+theme_classic()+ylim(-0.9,0.05)
GSEA_plots_H$`Palbo28 vs Cycling0`$`E2F TARGETS`+theme_classic()+ylim(-0.9,0.05)
dev.off()


# Get padj value of GSEA result for E2F TARGETS pathway to add onto GSEA plots
GSEA_padj <- list(DoxovsCyc=GSEA_Results_H$`Doxo14 vs Cycling0`@result[GSEA_Results_H$`Doxo14 vs Cycling0`@result$ID %in% "E2F TARGETS", ]$p.adjust, 
                  PalbovsCyc=GSEA_Results_H$`Palbo14 vs Cycling0`@result[GSEA_Results_H$`Palbo14 vs Cycling0`@result$ID %in% "E2F TARGETS", ]$p.adjust) # note that the padj values are really the same as the padj values as TNFA SIGNALING VIA NFKB. pvalues are different though. 
                  
GSEA_padj <- lapply(GSEA_padj, function(x){format(round(x, 4), nsmall = 4)}) # round to 4 decimal places 

p1 <- gseaplot(GSEA_Results_H$`Doxo14 vs Cycling0`, geneSetID = "E2F TARGETS", by="runningScore", title = "day 14 Doxo vs Cycling: E2F TARGETS", color.line="#B3C543", color="#B3C543", color.vline="#B3C543")+theme_classic()+ylim(-0.9, 0.05)+geom_text(label=paste0("padj=" ,GSEA_padj$DoxovsCyc), y=-0.6, x=5000)

p2 <- gseaplot(GSEA_Results_H$`Palbo14 vs Cycling0`, geneSetID = "E2F TARGETS", by="runningScore", title = "day 14 Palbo vs Cycling: E2F TARGETS", color="#E0ADCD", color.line="#E0ADCD", color.vline="#E0ADCD")+theme_classic()+ylim(-0.9, 0.05)+geom_text(label=paste0("padj=" ,GSEA_padj$PalbovsCyc), y=-0.6, x=5000)


pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs//GSEA_day14Doxo_day14Palbo_vs_Cyc_Hallmark_E2F_Targets.pdf", width=3, height=5)
plot_grid(p1, p2, ncol=1)
dev.off()
```

# RNA-seq: plot padj values for top 10 GO terms for shared down genes. 
```{r}
# Still showing the pvalues of the other clusters
GO_results_mat_clusters <- list()
for(i in 1:length(GO_results_list)){

GO_results_mat_clusters[[i]] <- GO_results_list_mat[row.names(GO_results_list_mat) %in% GO_results_list[[i]]$ID[1:25], ]
GO_results_mat_clusters[[i]] <- GO_results_mat_clusters[[i]][order(GO_results_mat_clusters[[i]][ ,i], decreasing=T), ]
}

pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/RNA_GO_results_SharedDown_genes.pdf", width=6, height=2.5)
pheatmap(GO_results_mat_clusters[[1]][1:10, ], cluster_cols = F, colorRampPalette(brewer.pal(n = 7, name ="Purples"))(100), breaks = seq(0,50,length.out = 100), cluster_rows = F, main="GO enrichment: shared up genes", annotation_row=GO_anno, annotation_col = Cluster_anno, annotation_colors = GO_anno_colors, border_color = "black")
dev.off()
```

# RNA-seq: plot padj values for top 5 IPA upstream regs for each cluster individually. 
```{r}
# Still showing the pvalues of the other clusters as columns but only the top IPA upstream regs from the cluster of interest. 
IPA_TF_regs_clusters <- list()
IPA_group_regs_clusters <- list()
for(i in 1:length(IPA_upstream_regs)){
  #transcription regulators
IPA_TF_regs_clusters[[i]] <-IPA_TF_regs_mat[row.names(IPA_TF_regs_mat) %in% IPA_upstream_regs[[i]]$Upstream.Regulator[grepl("transcription", IPA_upstream_regs[[i]]$Molecule.Type)][1:5], ]
IPA_TF_regs_clusters[[i]] <- IPA_TF_regs_clusters[[i]][order(IPA_TF_regs_clusters[[i]][ ,i], decreasing=T), ]
  # complex regulators
IPA_group_regs_clusters[[i]] <-IPA_group_regs_mat[row.names(IPA_group_regs_mat) %in% IPA_upstream_regs[[i]]$Upstream.Regulator[grepl("group", IPA_upstream_regs[[i]]$Molecule.Type)][1:5], ]
IPA_group_regs_clusters[[i]] <- IPA_group_regs_clusters[[i]][order(IPA_group_regs_clusters[[i]][ ,i], decreasing=T), ] # sort by decreasing padj value
}

# make heatmap of top 5 IPA predicted upstream regulators for shared up genes. 
pdf(file="/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/RNA_IPA_TF_regs_SharedDown_genes.pdf", width=3, height=2)
pheatmap(IPA_TF_regs_clusters[[1]], cluster_cols = F, colorRampPalette(brewer.pal(n = 7, name ="Greens"))(100), breaks = seq(0, 50,length.out = 100), cluster_rows = F, main="Shared down genes: predicted TF regulators", border_color = "black",  annotation_col=Cluster_anno, annotation_colors = GO_anno_colors)
dev.off()

pdf(file="/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/RNA_IPA_group_regs_SharedDown_genes.pdf", width=3.8, height=2)
pheatmap(IPA_group_regs_clusters[[1]], cluster_cols = F, colorRampPalette(brewer.pal(n = 7, name ="Greens"))(100), breaks = seq(0, 50,length.out = 100), cluster_rows = F, main="Shared down genes: predicted group regulators", border_color = "black", annotation_col=Cluster_anno, annotation_colors = GO_anno_colors)
dev.off()
```

# plot heatmap of E2F target genes, clustered based on log2FC from day 3 treatment controls & Cycling controls
```{r}
pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/RNA_log2FC_E2F_targets_mat.pdf", width=4.35, height=6)
pheatmap(CellCycle_sig_LFCMat, show_rownames = T, show_colnames=F, cluster_cols = F, cluster_rows=F, colorRampPalette(rev(brewer.pal(n = 7, name ="RdBu")))(100), breaks = seq(-6, 6,length.out = 100), annotation_row = clusterDF[ ,c("E2F", "Clusters")], annotation_col = anno_sigLFC, annotation_colors = hm_cols_RNA, border_color = NA, main="KEGG: Cell Cycle genes")
dev.off()
```

# plot log2FC of E2F target genes as violin plots across time
```{r}
E2F_genes <- clusterDF[clusterDF$E2F %in% "Yes", ]
E2F_genes <- row.names(E2F_genes[E2F_genes$Clusters %in% c("5", "2"), ])

log2FC_E2F <- rbind(
          data.frame(genes= E2F_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% E2F_genes, "Palbo3 vs Cycling0"], 
           Condition="day 3 Palbo", 
           Comparison="vs Cycling"),
  
          data.frame(genes= E2F_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% E2F_genes, "Palbo14 vs Cycling0"], 
           Condition="day 14 Palbo", 
           Comparison="vs Cycling" ), 
           
           data.frame(genes= E2F_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% E2F_genes, "Palbo21 vs Cycling0"], 
           Condition="day 21 Palbo", 
           Comparison="vs Cycling" ), 
           
           data.frame(genes= E2F_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% E2F_genes, "Palbo28 vs Cycling0"], 
           Condition="day 28 Palbo", 
           Comparison="vs Cycling" ),
           
           data.frame(genes= E2F_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% E2F_genes, "Doxo3 vs Cycling0"], 
           Condition="day 3 Doxo", 
           Comparison="vs Cycling"),
      
      data.frame(genes= E2F_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% E2F_genes, "Doxo14 vs Cycling0"], 
           Condition="day 14 Doxo", 
           Comparison="vs Cycling"), 
      
          data.frame(genes= E2F_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% E2F_genes, "Doxo21 vs Cycling0"], 
           Condition="day 21 Doxo", 
           Comparison="vs Cycling"), 
      
      data.frame(genes= E2F_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% E2F_genes, "Palbo14 vs Palbo3"], 
           Condition="day 14 Palbo",
           Comparison="vs day 3"), 
      
       data.frame(genes= E2F_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% E2F_genes, "Palbo21 vs Palbo3"], 
           Condition="day 21 Palbo",
           Comparison="vs day 3" ), 
           
           data.frame(genes= E2F_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% E2F_genes, "Palbo28 vs Palbo3"], 
           Condition="day 28 Palbo",
           Comparison="vs day 3" ),
      
      data.frame(genes= E2F_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% E2F_genes, "Doxo14 vs Doxo3"], 
           Condition="day 14 Doxo",
           Comparison="vs day 3"),
      
      data.frame(genes= E2F_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% E2F_genes, "Doxo21 vs Doxo3"], 
           Condition="day 21 Doxo",
           Comparison="vs day 3")
)
# function to create violin plot of log2FC values of genes
create_violin_plot <- function(data=log2FC_E2F, plot_title="RNA: log2FC of p53 target genes") {
  library(ggplot2)
  library(ggbeeswarm)
  
  ggplot(
    data,
    aes(x = Condition, y = log2FC, fill = Condition)
  ) +
    geom_violin() +
    geom_quasirandom(
      color = "black", alpha = 0.2,
      method = "pseudorandom", # Aligns jitter points to violin shape
      groupOnX = TRUE,
      size = 0.5
    ) +
    stat_summary(
      fun = "median",
      geom = "crossbar",
      color = "white",
      width = 0.5
    ) +
    theme_classic() +
    theme(
      axis.text.x = element_text(angle = 90, size = 8),
      axis.text.y = element_text(size = 8),
      axis.title.y = element_text(size = 10)
    ) +
    facet_wrap(~Comparison, scales = "free_x") +
    scale_fill_manual(
      values = c(
        "day 3 Doxo" = "#B3C543",
        "day 14 Doxo" = "#B3C543",
        "day 21 Doxo" = "#B3C543",
        "day 3 Palbo" = "#E0ADCD",
        "day 14 Palbo" = "#E0ADCD",
        "day 21 Palbo" = "#E0ADCD",
        "day 28 Palbo" = "#E0ADCD"
      )
    ) +
    ylab("log2FC") +
    xlab("") +
    ggtitle(plot_title)
}

pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/RNA_log2FC_E2F_targets.pdf", width=7, height=3)
library(ggbeeswarm)
create_violin_plot(
  data = log2FC_E2F,
  plot_title = "RNA: log2FC of E2F Target genes"
)
dev.off()

pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs/RNA_log2FC_day14_E2F_targets.pdf", width=4, height=3)
library(ggbeeswarm)
create_violin_plot(
  data = log2FC_E2F[log2FC_E2F$Condition %in% c("day 14 Palbo", "day 14 Doxo"), ],
  plot_title = "RNA: log2FC of E2F Target genes"
)
dev.off()


# get significance of log2FC values between day 14 Palbo & day 14 Doxo vs Cycling based on paired student t-test

# Check normality of differences 
d14_Palbo_log2FC <- log2FC_E2F[log2FC_E2F$Condition %in% "day 14 Palbo" & log2FC_E2F$Comparison %in% "vs Cycling", ]$log2FC # log2FC values of p53 targets in day 14 Palbo vs Cycling

d14_Doxo_log2FC <- log2FC_E2F[log2FC_E2F$Condition %in% "day 14 Doxo" & log2FC_E2F$Comparison %in% "vs Cycling", ]$log2FC # log2FC values of p53 targets in day 14 Doxo vs Cycling

differences <- d14_Palbo_log2FC - d14_Doxo_log2FC

# Shapiro-Wilk test for normality
shapiro_test <- shapiro.test(differences)
print(shapiro_test)

# Visualize distribution of differences
ggplot(data.frame(differences), aes(x = differences)) +
  geom_histogram(binwidth = 0.2, fill = "skyblue", color = "black", alpha = 0.7) +
  labs(title = "Distribution of Log2FC Differences", x = "Difference (Drug1 - Drug2)", y = "Count")

# Perform Paired t-test if normality assumption holds
if (shapiro_test$p.value > 0.05) {
  t_test_result <- t.test(d14_Palbo_log2FC, d14_Doxo_log2FC, paired = TRUE)
  print(t_test_result)
} else {
  # Use Wilcoxon signed-rank test if normality is violated
  wilcox_test_result <- wilcox.test(d14_Palbo_log2FC, d14_Doxo_log2FC, paired = TRUE)
  print(wilcox_test_result)
}
```


# ATAC-seq: plot motif enrichment results in the E2F family for down peaks across all timepoints and drug treatments (DESeq comparisons)
```{r}
# load in function to generate matrix of -log10(padj) values of motif enrichment results
load("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Supplemental_Figure_Summary_Motif_enrichment_results_all_sig_peaks/Functions/create_AME_peaks_matrix.RData")

# load in function to create dotplots for motif enrichment results for each DESeq comparison. 
load("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Supplemental_Figure_Summary_Motif_enrichment_results_all_sig_peaks/Functions/create_AME_dotplots.RData")

E2F_DownPeaks_AME_mat <- create_AME_peaks_matrix(annoDownPeaks_AME_df[grepl("E2F", annoDownPeaks_AME_df$motif_ID), ])
                                                      
E2F_DownPeaks_AME_mat <- E2F_DownPeaks_AME_mat[order(rowSums(E2F_DownPeaks_AME_mat), decreasing=T), ] # order by overall most to least enriched E2F motifs 

outdir <- "/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Cell_Cycle_Arrest_Signature_in_Both_Drugs"

# plot motif enrichment values of E2F motifs in all down peaks for all timepoints vs Cycling. 
pdf(paste0(outdir, "/ATAC_E2F_DownPeaks_AME_dotplots.pdf"), width=6, height=4)
create_AME_dotplots(annoDownPeaks_AME_df[grepl("E2F", annoDownPeaks_AME_df$motif_ID) & grepl("vsCycling", annoDownPeaks_AME_df$Comparison), ], Title="E2F Motifs enriched in Down peaks", fill_limit=200, size_limit=51)
dev.off()

# note: initially, I considered only showing E2F motif enrichment nearby shared down genes (enriched in cell cycle related genes) but E2F appears to also be enriched nearby genes going up with drug treatment. Therefore, I think it makes sense to just show the motif enrichment in all down peaks since the enrichment is not dependent/exclusive on how close it is to shared down genes
```


```{r}
# write function to create venn diagrams that are proportional to size of each category. 
create_proportional_venn <- function(sets, set_labels, colors = c("#E0ADCD", "#B3C543", "#A0ADCD")) {
  library(eulerr)
  # Check if sets and labels are provided correctly
  if (length(sets) != length(set_labels)) {
    stop("The number of sets and set_labels must be the same.")
  }
  
  # Prepare the data for eulerr
  venn_data <- sets
  names(venn_data) <- set_labels
  
  # Use eulerr to calculate the Venn diagram
  venn_fit <- euler(venn_data)
  
  # Plot the Venn diagram
  plot(venn_fit, 
       fills = colors, 
       labels = list(cex = 1, fontfamily = "Helvetica"),
       quantities = list(cex = 1),
       edges = list(lwd = 2))
}

# unique peak ids for Doxo vs Cycling down tp peaks with E2F1 motif
tp_downpeaks_Doxo <- unique(unlist(E2F1_tp_peaks[grepl("Doxo", names(E2F1_tp_peaks))]))

# unique peak ids for Palbo vs Cycling down tp peaks with E2F1 motif
tp_downpeaks_Palbo <- unique(unlist(E2F1_tp_peaks[grepl("Palbo", names(E2F1_tp_peaks))]))

# peak ids for all down peaks 
all_downpeaks_ids <- unique(c(mergedDiffGR$allDown_Doxo$name, mergedDiffGR$allDown_Palbo$name))

# plot venn diagram showing overlap of E2F1 containing down peaks between Doxo & Palbo against number of all down peaks
pdf(paste0(outdir, "/ATAC_E2F1_tp_downpeaks_PalbovsDoxo_overlap_venn.pdf"), width=6, height=4)
create_proportional_venn(sets=list(Palbo=tp_downpeaks_Palbo, Doxo=tp_downpeaks_Doxo, allDown=all_downpeaks_ids), set_labels = c("Palbo", "Doxo", "all Down"), colors = c("#E0ADCD", "#B3C543", "#F6F2EA"))
dev.off()

# unique peak ids for Doxo vs Cycling down tp peaks with any E2F motif
tp_downpeaks_Doxo <- unique(unlist(all_E2F_tp_peaks[grepl("Doxo", names(all_E2F_tp_peaks))]))

# unique peak ids for Palbo vs Cycling down tp peaks with any E2F motif
tp_downpeaks_Palbo <- unique(unlist(all_E2F_tp_peaks[grepl("Palbo", names(all_E2F_tp_peaks))]))

# plot venn diagram showing overlap of all E2F containing down peaks between Doxo & Palbo against number of all down peaks
pdf(paste0(outdir, "/ATAC_all_E2F_tp_downpeaks_PalbovsDoxo_overlap_venn.pdf"), width=6, height=4)
create_proportional_venn(sets=list(Palbo=tp_downpeaks_Palbo, Doxo=tp_downpeaks_Doxo, allDown=all_downpeaks_ids), set_labels = c("Palbo", "Doxo", "all Down"), colors = c("#E0ADCD", "#B3C543", "#F6F2EA"))
dev.off()
```
