---
title: "Figure p53 activity is different depending on the presence of DNA damage"
author: "Joanna Yeung"
date: '2025-01-24'
output: html_document
---
```{r}
library(readr)
library(GenomicFeatures)
library(DESeq2)
library(apeglm)
library(gridExtra)
library(ggplot2)
library(pheatmap)
library(msigdbr)
library(stringr)
library(org.Hs.eg.db)
library(ggrepel)
library(rtracklayer)
library(cowplot)
library(readxl)
library(EnsDb.Hsapiens.v79)
library(viridis)
library(GenomicRanges)
library(rtracklayer)
library(RColorBrewer)
library(cowplot)
library(preprocessCore)
library(scales)
library(ChIPseeker)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(org.Hs.eg.db)
library(BSgenome.Hsapiens.UCSC.hg38)
library(MotifDb)
library(motifmatchr)
library(Biostrings)
library(clusterProfiler)
library(pheatmap)
```

# load in workspace with DESeq results and rlog counts matrix of genes. 
```{r}
load("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_Global_transcriptome_and_chromatin_accessibility_is_dynamic_in_both_types_of_therapy_induced_senescence/workspace.RData")
```

#### universe (background gene set) for GO enrichment analysis is all genes expressed
```{r}
universe <- genes_anno_type[match(gsub('\\.[0-9]*$', '', row.names(acrossGroups)), genes_anno_type$ensembl_id), ]
```

# GO enrichment analysis on k-means clusters from sigMat (significantly changing genes across all conditions, based on Likelihood ratio test)

## *function to get GO enriched terms*:
## arguments:
### gene= vector of genes to test for GO enrichment (gene ensembl format)
### universe=background gene set 
### Database=database containing groups of genes to GO annotation term e.g. msigdb GO:BP
```{r}
getGOenrichment <-function(gene,universe,Database){
  #gene_entrezID <-  AnnotationDbi::select(EnsDb.Hsapiens.v79, gene[!is.na(gene)], "ENTREZID", "SYMBOL") #convert gene IDs from SYMBOL format to ENTREZ ID format (was using v79, new code is using v107)
  gene_entrezID <- genes_anno_type[match(gene, genes_anno_type$gene_id), ]
  gene_go <- enricher(gene_entrezID$entrezid, TERM2GENE=Database, universe=universe) #enrichment analysis with hypergeometric test
  gene_go <- setReadable(gene_go, OrgDb = org.Hs.eg.db, keyType="ENTREZID") #convert enrichment results from ENTREZ ID format to SYMBOL for easier interpretation
return(gene_go)
} 
```

# assign significant genes to relevant k-means clusters & drug + cell cycle arrest categories. 
```{r}
genes_in_clusters <- lapply(levels(clusterDF$Clusters), function(x){row.names(clusterDF[clusterDF$Clusters %in% x, ])})
genes_in_clusters[[1]] <- unlist(genes_in_clusters[1:2])
genes_in_clusters[[6]] <- unlist(genes_in_clusters[5:6])
genes_in_clusters <- genes_in_clusters[c(1,3,4,6,7,8)]
cluster_names <- c("CellCycle", "Palbo1", "Palbo2", "Doxo1", "Doxo2", "Shared") 
names(genes_in_clusters) <- cluster_names
```

# RNA-seq: GO enrichment on significant genes
```{r}
# GO enrichment analysis with msigdb GO:BP database:
GOBP_enrichment_clusters <- list()
# GO enrichment analysis with msigdb REACT database:
GO_enrichment_clusters_REACT <- list()
# GO enrichment analysis with msigdb Hallmark database:
GO_enrichment_clusters_HallMark <- list()
# GO enrichment analysis with msigdb REACT database:
GO_enrichment_clusters_REACT <- list()

# GO enrichment analysis with msigdb GO:MF database:
GO_enrichment_clusters_MF <- list()

for(i in 1:length(genes_in_clusters)){
  
GOBP_enrichment_clusters[[i]] <- getGOenrichment(genes_in_clusters[[i]], as.character(universe$entrezid),msigdbr_t2g_GO_BP) # do GO enrichment analysis with background gene set set as all gene expressed, database used is msigdb GO:BP

GO_enrichment_clusters_REACT[[i]] <- getGOenrichment(genes_in_clusters[[i]], as.character(universe$entrezid),msigdbr_t2g_REACT)

GO_enrichment_clusters_HallMark[[i]] <- getGOenrichment(genes_in_clusters[[i]], as.character(universe$entrezid),msigdbr_t2g_H)

GO_enrichment_clusters_REACT[[i]] <- getGOenrichment(genes_in_clusters[[i]], as.character(universe$entrezid),msigdbr_t2g_REACT)

 GO_enrichment_clusters_MF[[i]] <- getGOenrichment(genes_in_clusters[[i]], as.character(universe$entrezid),msigdbr_t2g_GO_MF)
}
```

### RNA-seq: format GO enrichment terms from each cluster into matrix of -log10(padj values)
```{r}
# convert each clusters' GO enrichment terms into dataframes
GO_enrichment_clusters_KEGG_df <- list()
 GO_enrichment_clusters_REACT_df <- list()
 GO_enrichment_clusters_HallMark_df <- list()
 GOBP_enrichment_clusters_df <- list()
for(i in 1:length(GO_enrichment_clusters_REACT)){
  GO_enrichment_clusters_REACT_df[[i]] <- as.data.frame(GO_enrichment_clusters_REACT[[i]])
   GO_enrichment_clusters_REACT_df[[i]] <- as.data.frame(GO_enrichment_clusters_REACT[[i]])
 GO_enrichment_clusters_HallMark_df[[i]] <- as.data.frame(GO_enrichment_clusters_HallMark[[i]])
 GOBP_enrichment_clusters_df[[i]] <- as.data.frame(GOBP_enrichment_clusters[[i]])
}

 # indicate names of each list as the cluster number
 names(GO_enrichment_clusters_KEGG_df) <- cluster_names 
 names(GO_enrichment_clusters_REACT_df) <- cluster_names 
 names(GO_enrichment_clusters_HallMark_df) <- cluster_names 
 names(GOBP_enrichment_clusters_df) <- cluster_names 
 
 # remove any cluster of genes that have 0 GO enrichment terms
 GO_enrichment_clusters_KEGG_df <-  GO_enrichment_clusters_KEGG_df[sapply( GO_enrichment_clusters_KEGG_df, nrow)>0]
 GO_enrichment_clusters_REACT_df <-  GO_enrichment_clusters_REACT_df[sapply( GO_enrichment_clusters_REACT_df, nrow)>0]
 GO_enrichment_clusters_HallMark_df <-  GO_enrichment_clusters_HallMark_df[sapply( GO_enrichment_clusters_HallMark_df, nrow)>0]
 GOBP_enrichment_clusters_df <-  GOBP_enrichment_clusters_df[sapply( GOBP_enrichment_clusters_df, nrow)>0]
 
 # indicate which cluster and database each dataframe comes from 
 for(i in 1:length(GO_enrichment_clusters_KEGG_df)){
  GO_enrichment_clusters_KEGG_df[[i]]$Cluster <- names(GO_enrichment_clusters_KEGG_df)[i]
  GO_enrichment_clusters_KEGG_df[[i]]$Database <- "KEGG"
 }
  for(i in 1:length(GO_enrichment_clusters_REACT_df)){
   GO_enrichment_clusters_REACT_df[[i]]$Cluster <-  names(GO_enrichment_clusters_REACT_df)[i]
   GO_enrichment_clusters_REACT_df[[i]]$Database <- "REACTOME"
  }
  for(i in 1:length(GO_enrichment_clusters_HallMark_df)){
 GO_enrichment_clusters_HallMark_df[[i]]$Cluster <- names(GO_enrichment_clusters_HallMark_df)[i]
 GO_enrichment_clusters_HallMark_df[[i]]$Database <- "Hallmark"
  }
 
  for(i in 1:length(GOBP_enrichment_clusters_df)){
 GOBP_enrichment_clusters_df[[i]]$Cluster <- names(GOBP_enrichment_clusters_df)[i]
  GOBP_enrichment_clusters_df[[i]]$Database <- "GO:BP"
  }

# combine GO enrichment results from different databases into 1 dataframe for each cluster of genes. 
GO_results_list <- vector()
GO_results_list$CellCycle <- rbind(GO_enrichment_clusters_KEGG_df$CellCycle, GO_enrichment_clusters_HallMark_df$CellCycle, GO_enrichment_clusters_REACT_df$CellCycle, GOBP_enrichment_clusters_df$CellCycle)

GO_results_list$Palbo1 <- rbind(GO_enrichment_clusters_KEGG_df$Palbo1, GO_enrichment_clusters_HallMark_df$Palbo1 , GO_enrichment_clusters_REACT_df$Palbo1, GOBP_enrichment_clusters_df$Palbo1)

GO_results_list$Palbo2 <- rbind(GO_enrichment_clusters_KEGG_df$Palbo2, GO_enrichment_clusters_HallMark_df$Palbo2, GO_enrichment_clusters_REACT_df$Palbo2, GOBP_enrichment_clusters_df$Palbo2)

GO_results_list$Doxo1 <- rbind(GO_enrichment_clusters_KEGG_df$Doxo1, GO_enrichment_clusters_HallMark_df$Doxo1, GO_enrichment_clusters_REACT_df$Doxo1, GOBP_enrichment_clusters_df$Doxo1)

GO_results_list$Doxo2 <- rbind(GO_enrichment_clusters_KEGG_df$Doxo2, GO_enrichment_clusters_HallMark_df$Doxo2, GO_enrichment_clusters_REACT_df$Doxo2, GOBP_enrichment_clusters_df$Doxo2)

GO_results_list$Shared <- rbind(GO_enrichment_clusters_KEGG_df$Shared, GO_enrichment_clusters_HallMark_df$Shared, GO_enrichment_clusters_REACT_df$Shared, GOBP_enrichment_clusters_df$Shared)

# order GO enrichment results from different databases by padj values
for(i in 1:length(GO_results_list)){
  GO_results_list[[i]] <- GO_results_list[[i]][order(GO_results_list[[i]]$p.adjust), ]
}

# combine dataframes of each cluster of gene's GO enrichment results into 1 master dataframe
GO_results_df <- do.call(rbind, GO_results_list)

# create matrix of -log10pvalues of the top 15 enriched pathways for each cluster

GO_results_list_mat <- matrix(nrow=length(unique(unlist(lapply(GO_results_list, function(GO_results_list){GO_results_list$ID[1:50]})))), ncol=length(GO_results_list)) # make empty matrix, number of columns are the clusters of genes and number of total top GO terms of all clusters

row.names(GO_results_list_mat) <- unique(unlist(lapply(GO_results_list, function(GO_results_list){GO_results_list$ID[1:50]}))) # row names are top 12 GO enrichment terms for each cluster combined
colnames(GO_results_list_mat) <- cluster_names  # column names are clusters

# pathways that are not enriched in the cluster are listed as NA so just change the value to 0 so that it can be plotted in heatmap
for(i in 1:ncol(GO_results_list_mat)){
GO_results_list_mat[ ,i] <- -log10(GO_results_list[[i]][match(row.names(GO_results_list_mat), GO_results_list[[i]]$ID), ]$p.adjust)
GO_results_list_mat[ ,i] <- ifelse(GO_results_list_mat[ ,i] %in% NA, 0, GO_results_list_mat[ ,i])
}
GO_results_list_mat <- GO_results_list_mat[!is.na(row.names(GO_results_list_mat)), ] # remove padj values that are NA 

GO_anno_colors <-  list(database=c("GO:BP"="#3B4992FF", "Hallmark"="#EE0000FF", "KEGG"="#008B45FF", "REACTOME" ="#631879FF"), Cluster=c("CellCycle"="#66C2A5", "Palbo1"="#FC8D62", "Palbo2"="#8DA0CB","Doxo1"="#E78AC3","Doxo2"="#FFD92F", "Shared"="#E5C494"))
names(GO_anno_colors$Cluster) <- cluster_names 
# annotation dataframe for rows of heatmap to indicate which database GO term came from 
GO_anno <- data.frame(row.names=row.names(GO_results_list_mat), database=GO_results_df[match(row.names(GO_results_list_mat), GO_results_df$ID), ]$Database)
# annotation dataframe for columns of heatmap to color by Cluster colors. 
Cluster_anno <- data.frame(Cluster=cluster_names , row.names = colnames(GO_results_list_mat))
```
# RNA-seq: GSEA Analysis
```{r}
ranked_res_forGSEA <- list()
for(i in 1:length(LFCResultsall_RNA)){
  
  rank <- LFCResultsall_RNA[[i]]
  
  # Assign a default p-value for genes with NA padj (maximum p-value)
  rank$pvalue[is.na(rank$padj)] <- 1 
  
  # Create a ranked list based on log2FoldChange and -log10(p-value)
  rank$rank_metric <- sign(rank$log2FoldChange) * -log10(rank$pvalue)
  
  # Sort by the ranking metric
  rank <- rank[order(rank$rank_metric, decreasing = TRUE), ]
  
  # Store the ranked values
  ranked_res_forGSEA[[i]] <- rank$rank_metric
  names(ranked_res_forGSEA[[i]]) <- row.names(rank)

  # Convert gene IDs to Entrez IDs if available
  entrez_ids <- genes_anno_type[match(names(ranked_res_forGSEA[[i]]), genes_anno_type$ensembl_id), ]$entrezid
  names(ranked_res_forGSEA[[i]]) <- ifelse(is.na(entrez_ids), names(ranked_res_forGSEA[[i]]), entrez_ids)
}

names(ranked_res_forGSEA) <- names(LFCResultsall_RNA)


# run GSEA analysis using: 
GSEA_Results_H <- list() # msigdb Hallmark database
GSEA_Results_REACT <- list() # msigdb KEGG database
GSEA_Results_REACT <- list() # msigdb Reactome database
GSEA_plots_H <- list()
GSEA_plots_REACT <- list()
GSEA_plots_REACT <- list()
for(i in 1:length(ranked_res_forGSEA)){
 GSEA_Results_H[[i]] <- GSEA(ranked_res_forGSEA[[i]], TERM2GENE = msigdbr_t2g_H) 
 GSEA_Results_REACT[[i]] <- GSEA(ranked_res_forGSEA[[i]], TERM2GENE = msigdbr_t2g_REACT) 
 GSEA_Results_REACT[[i]] <- GSEA(ranked_res_forGSEA[[i]], TERM2GENE = msigdbr_t2g_REACT) 
}

names(GSEA_Results_H) <- names(LFCResultsall_RNA)
names(GSEA_Results_REACT) <- names(LFCResultsall_RNA)
names(GSEA_Results_REACT) <- names(LFCResultsall_RNA)

# generate running score plots for GSEA results
# GSEA Hallmark
for(i in 1:length(GSEA_Results_H)){
  maxplots <- length(GSEA_Results_H[[i]]$ID)
GSEA_plots_H[[i]] <- list()
GSEA_plots_KEGG[[i]] <- list()
GSEA_plots_REACT[[i]] <- list()
  for(j in 1:maxplots){
  GSEA_plots_H[[i]][[j]] <- gseaplot(GSEA_Results_H[[i]], geneSetID = j, by = "runningScore", title = paste0(names(GSEA_Results_H)[i], ": " , GSEA_Results_H[[i]]$ID[j]) )
  }
  names(GSEA_plots_H[[i]]) <- GSEA_Results_H[[i]]$ID[1:maxplots]
}

# GSEA KEGG
for (i in 1:length(GSEA_Results_KEGG)) {
  # Check if the current object is not empty
  if (!is.null(GSEA_Results_KEGG[[i]]) && length(GSEA_Results_KEGG[[i]]$ID) > 0) {
    maxplots <- length(GSEA_Results_KEGG[[i]]$ID)
    GSEA_plots_KEGG[[i]] <- list()  # Initialize a list to store plots for this object
    
    for (j in 1:maxplots) {
      GSEA_plots_KEGG[[i]][[j]] <- gseaplot(
        GSEA_Results_KEGG[[i]], 
        geneSetID = j, 
        by = "runningScore", 
        title = GSEA_Results_KEGG[[i]]$ID[j]
      )
    }
    # Assign names to the plots
    names(GSEA_plots_KEGG[[i]]) <- GSEA_Results_KEGG[[i]]$ID[1:maxplots]
  } else {
    # If empty, set the current slot in GSEA_plots_KEGG to NULL
    GSEA_plots_KEGG[[i]] <- NULL
  }
}

# GSEA Reactome
for (i in 1:length(GSEA_Results_REACT)) {
  # Check if the current object is not empty
  if (!is.null(GSEA_Results_REACT[[i]]) && length(GSEA_Results_REACT[[i]]$ID) > 0) {
    maxplots <- length(GSEA_Results_REACT[[i]]$ID)
    GSEA_plots_REACT[[i]] <- list()  # Initialize a list to store plots for this object
    
    for (j in 1:maxplots) {
      GSEA_plots_REACT[[i]][[j]] <- gseaplot(
        GSEA_Results_REACT[[i]], 
        geneSetID = j, 
        by = "runningScore", 
        title = GSEA_Results_REACT[[i]]$ID[j]
      )
    }
    # Assign names to the plots
    names(GSEA_plots_REACT[[i]]) <- GSEA_Results_REACT[[i]]$ID[1:maxplots]
  } else {
    # If empty, set the current slot in GSEA_plots_REACT to NULL
    GSEA_plots_REACT[[i]] <- NULL
  }
}

# assign names to plots based on which DESeq comparison GSEA was done on. 
names(GSEA_plots_H) <- names(LFCResultsall_RNA)
names(GSEA_plots_KEGG) <- names(which(unlist(lapply(GSEA_Results_KEGG, function(x){length(x$ID)})) >0))
names(GSEA_plots_REACT) <- names(which(unlist(lapply(GSEA_Results_REACT, function(x){length(x$ID)})) >0))
```

# RNA-seq: save GSEA running score plots to pdf
```{r}
pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/GSEA_plots_Hallmark.pdf", width=5, height=4)
GSEA_plots_H
dev.off()
```

# # RNA-seq: generate input files for IPA pathways. 
```{r}
# clusters of genes 
for(i in 1:length(genes_in_clusters)){
  
  IPA_input <- as.data.frame(genes_in_clusters[[i]])
  colnames(IPA_input) <- "ID"
  write.table(IPA_input, file=paste0("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/IPA_files/", cluster_names[i], ".csv"), row.names=F, sep=",")
}
# DESeq Wald Test results for pairwise comparisons. 
for(i in 1:length(LFCResultsall_RNA)){
  write.table(as.data.frame(LFCResultsall_RNA[[i]]), file=paste0("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/IPA_files/DESeq_results_input/", names(LFCResultsall_RNA)[i], ".csv"), sep=",")
}
```

# # RNA-seq: import IPA upstream regulators & pathways
```{r}
setwd("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/IPA_files/output/")
IPA_upstream_regs <- list()
IPA_pathways <- list()
# read in information for enriched IPA pathways per hierarchical cluster (clustering based on scaled rlog normalized counts)
for(i in 1:length(cluster_names)){
IPA_pathways[[i]]<- read_tsv(paste0(cluster_names[i], "_IPA_Pathways.txt"), skip=2)
# read in information for enriched upstream regulators 
IPA_upstream_regs[[i]] <- read.table(paste0(cluster_names[i], "_IPA_UpstreamRegs.txt"), sep="\t", skip=2, header=T)
IPA_upstream_regs[[i]] <- IPA_upstream_regs[[i]][!grepl("mir|miR|Mir", IPA_upstream_regs[[i]]$Upstream.Regulator), ] # remove upstream regulators that are miRNAs (uncertain if they will be informative & there is a lot of them)
}

# filter out upstream regulators that are not expressed in this cell line. 

#extract ids of all expressed genes
expressedGenes <- LFCResultsall_RNA[[1]]$SYMBOL

# keep only upstream regulators that are part of expressed genes or have lower case letters (these are upstream regulators that could be miRNAs or protein complexes)
IPA_upstream_regs <- lapply(IPA_upstream_regs, function(IPA_upstream_regs){IPA_upstream_regs[c(grep('[:lower:]', IPA_upstream_regs$Upstream.Regulator), which(IPA_upstream_regs$Upstream.Regulator %in% expressedGenes)), ]})

names(IPA_upstream_regs) <- cluster_names

names(IPA_pathways) <- cluster_names
```

# RNA-seq: make matrix with log2FC values of significant genes
```{r}
# get log2FC values of significantly changing genes over time
sig_LFCMat <- matrix(ncol=length(LFCResultsall_RNA), nrow=nrow(clusterDF))
colnames(sig_LFCMat) <- names(LFCResultsall_RNA)
row.names(sig_LFCMat) <-row.names(LFCResultsall_RNA[[10]][row.names(LFCResultsall_RNA[[10]]) %in% sigGenes, ])
for(i in 1:ncol(sig_LFCMat)){
sig_LFCMat[ ,i] <- LFCResultsall_RNA[[i]][row.names(LFCResultsall_RNA[[i]]) %in% sigGenes, ]$log2FoldChange
}

row.names(sig_LFCMat) <-  ifelse(genes_anno_type[match(row.names(sig_LFCMat), genes_anno_type$ensembl_id), ]$gene_id %in% NA, row.names(sig_LFCMat), genes_anno_type[match(row.names(sig_LFCMat), genes_anno_type$ensembl_id), ]$gene_id)

row.names(ordered_sigMat)[grep("^H1.[0-9]", row.names(ordered_sigMat))]  <- gsub("*H1.", "H1-",row.names(ordered_sigMat[grepl("^H1.[0-9]", row.names(ordered_sigMat)), ]) )

sig_LFCMat <- sig_LFCMat[match(row.names(ordered_sigMat), row.names(sig_LFCMat)), ] # order based on k-means & hierarchical ordering of rlog counts. 

# get log2FC values of all expressed genes over time 
LFCMat <- matrix(ncol=length(LFCResultsall_RNA), nrow=nrow(LFCResultsall_RNA[[1]]))
colnames(LFCMat) <- names(LFCResultsall_RNA)
row.names(LFCMat) <-row.names(LFCResultsall_RNA[[1]])
for(i in 1:ncol(LFCMat)){
LFCMat[ ,i] <- LFCResultsall_RNA[[i]]$log2FoldChange
}

row.names(LFCMat) <-  ifelse(genes_anno_type[match(row.names(LFCMat), genes_anno_type$ensembl_id), ]$gene_id %in% NA, row.names(LFCMat), genes_anno_type[match(row.names(LFCMat), genes_anno_type$ensembl_id), ]$gene_id)
```

# RNA-seq: P53 & NF-kB signatures
```{r}
# import p53 census target gene database from https://tp53.isb-cgc.org/target_genes
Fischer_Oncogene_2017_p53_census_target_genes <- read_excel("/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/RNAseq/external_databases/Fischer_Oncogene_2017_p53_census_target_genes.xlsx")

# import Gilmore Lab's list of NF-kB target genes
Gilmore_NFKB_Targets <- as.data.frame(read_excel("/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/RNAseq/external_databases/GilmoreLab_NFKB_Target_Genes_BostonUniversity.xlsx"))$`Human Gene Name`
Gilmore_NFKB_Targets <- unique(Gilmore_NFKB_Targets[!is.na(Gilmore_NFKB_Targets)])

# extract genes under Hallmark terms P53 PATHWAY & TNFA SIGNALING VIA NFKB
p53_Hallmark <- msigdbr_df_H[msigdbr_df_H$gs_name %in% "HALLMARK_P53_PATHWAY", ]$gene_symbol
NFKB_Hallmark <- msigdbr_df_H[msigdbr_df_H$gs_name %in% "HALLMARK_TNFA_SIGNALING_VIA_NFKB", ]$gene_symbol

# filter for p53 & NFKB target genes from IPA
IPA_targets <- list(p53=unlist(lapply(IPA_upstream_regs, function(x){str_split(x[grep("TP53$", x$Upstream.Regulator), ]$Target.Molecules.in.Dataset, ",")})),
                    NFKB=unlist(lapply(IPA_upstream_regs, function(x){unique(unlist(str_split(x[grep("NFKB", x$Upstream.Regulator), ]$Target.Molecules.in.Dataset, ",")))}))
                    )

# make log2FC matrix of p53 target genes going up with drug treatment
p53_sig_LFCMat <- sig_LFCMat[row.names(sig_LFCMat) %in% c(Fischer_Oncogene_2017_p53_census_target_genes$`Gene Symbol`, p53_Hallmark) & ! row.names(sig_LFCMat) %in% row.names(clusterDF[clusterDF$Clusters %in% c("5", "2"), ]), ]

anno_sigLFC <- data.frame(
  row.names = colnames(sig_LFCMat),
  log2FC = c(rep("vs Cyc", 4), rep("vs day3", 3), rep("vs Cyc", 5), rep("vs day3", 4), rep("DoxovsPalbo",4)),
  Treatment = c(rep("Doxo", 7), rep("Palbo", 9), rep("DoxovsPalbo",4)),
  Timepoint = c("3", "10", "14", "21", "10", "14", "21", "3", "10", "14", "21", "28", "10", "14", "21", "28","3", "10", "14", "21")
)

# add annotation to significant genes based on if they are NF-kB or p53 targets. 
clusterDF$NFKB_All <- ifelse(row.names(clusterDF) %in% c(Gilmore_NFKB_Targets, NFKB_Hallmark, IPA_targets$NFKB), "Yes", "No") # annotate sig genes as NF-kB targets if they are part of Gilmore_NFKB_Targets or TNFA SIGNALING VIA NFKB Hallmark term 
clusterDF$p53_All <- ifelse(row.names(clusterDF) %in% c(Fischer_Oncogene_2017_p53_census_target_genes$`Gene Symbol`, p53_Hallmark), "Yes", "No")

row.names(clusterDF)[grep("^H1.[0-9]", row.names(clusterDF))]  <- gsub("*H1.", "H1-",row.names(clusterDF[grepl("^H1.[0-9]", row.names(clusterDF)), ]) ) # correction: to match with row.names in sig_LFCMat, replace - with . for linker histone genes. 
```


# make matrix of normalized counts across drug & time. 
```{r}
NormCountsMat <- counts(RNAddsall, normalized=T)
# convert row names of matrix of normalized counts of off to on genes to gene symbol format. If there is no gene symbol format, keep ensembl format. 
row.names(NormCountsMat) <- gsub('\\.[0-9]*$', '', row.names(NormCountsMat))
row.names(NormCountsMat) <- ifelse(genes_anno_type[match(row.names(NormCountsMat), genes_anno_type$ensembl_id), ]$gene_id %in% NA, row.names(NormCountsMat), genes_anno_type[match(row.names(NormCountsMat), genes_anno_type$ensembl_id), ]$gene_id)

# convert matrix of normalized counts of off to on genes into a dataframe with metaData information for ggplot to accept
NormCounts_df <- list()
for(i in 1:nrow(NormCountsMat)){
NormCounts_df[[i]] <- data.frame(Counts=NormCountsMat[i, ], sample=names((NormCountsMat[i, ])), Transcript= row.names(NormCountsMat)[i], metaDataall)
}
NormCounts_df <- do.call(rbind, NormCounts_df) # combine list into 1 giant dataframe
```


# ATAC-seq: plot Tn5 insertion sites around p53 motifs overlapping up peaks to show stronger p53 chromatin accessibility signature in Doxo compared to Palbo. 
```{r}
# filter for GRanges of up peaks in each drug and cell cycle arrest category 
Up_DrugandArrestCat <- list(
  PalboUpvsCyc=allUpvsCycling$Palbo[!allUpvsCycling$Palbo %over% allUpvsCycling$Doxo, ], 
  sharedUpvsCyc=allUpvsCycling$Palbo[allUpvsCycling$Palbo %over% allUpvsCycling$Doxo, ],
  DoxoUpvsCyc=allUpvsCycling$Doxo[!allUpvsCycling$Doxo %over% allUpvsCycling$Palbo, ], 
  PalboUpvsday3=allUpvsday3$Palbo[!allUpvsday3$Palbo %over% allUpvsday3$Doxo, ], 
  sharedUpvsday3=allUpvsday3$Palbo[allUpvsday3$Palbo %over% allUpvsday3$Doxo, ], 
  DoxoUpvsday3=allUpvsday3$Doxo[!allUpvsday3$Doxo %over% allUpvsday3$Palbo, ])

# filter for GRanges of up peaks in each drug and cell cycle arrest category but without specifying shared up peaks. 
Up_DrugandArrest_PvsD <- list(
  PalboUpvsCyc=allUpvsCycling$Palbo, 
  DoxoUpvsCyc=allUpvsCycling$Doxo, 
  PalboUpvsday3=allUpvsday3$Palbo,
  DoxoUpvsday3=allUpvsday3$Doxo)

# load in HOCOMOCO motif sequence database
load("/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/post_peakcalling_analysis/motifs/HOCOMOCO_v11_matrix.RData")
# scan for p53 motifs
p53_HMv11 <- matchPWM(HOCOMOCO_v11_matrix$P53_HUMAN.H11MO.0.A, BSgenome.Hsapiens.UCSC.hg38, with.score=T)

# filter for p53 motifs that overlap with up peaks
p53_ATAC_Up_DrugandArrestCat <- lapply(Up_DrugandArrestCat, function(x){p53_HMv11[p53_HMv11 %over% x]})
p53_ATAC_Up_DrugandArrest_PvsD <- lapply(Up_DrugandArrest_PvsD, function(x){p53_HMv11[p53_HMv11 %over% x]})

# Export each GRanges object as a BED file
for(i in 1:length(p53_ATAC_Up_DrugandArrestCat)){
export.bed(p53_ATAC_Up_DrugandArrestCat[[i]], con = paste0("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/deepTools_files/input_beds/", names(p53_ATAC_Up_DrugandArrestCat)[i], "_p53_HMv11.bed"))
}

for(i in 1:length(p53_ATAC_Up_DrugandArrest_PvsD)){
export.bed(p53_ATAC_Up_DrugandArrest_PvsD[[i]], con = paste0("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/deepTools_files/input_beds/", names(p53_ATAC_Up_DrugandArrest_PvsD)[i], "_PvsD_p53_HMv11.bed"))
}

# generate input files for deepTools scripts
# directory paths to bed files 
writeLines(dir("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/deepTools_files/input_beds", full.names=T, pattern=".bed"), con="/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/deepTools_files/input_beds/regions.txt")

# filenames to be assigned to plots. 
filenames <- gsub(".bed", "", dir("//lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/deepTools_files/input_beds", pattern=".bed"))
writeLines(filenames, con="/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/deepTools_files/input_beds/filename.txt")
```

# submit bash job to plot around Tn5 cut sites around p53 motifs in up peaks
```{bash}
mergedcuts="/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/Cyclingmerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day3_Doxomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day10_Doxomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day14_Doxomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day21_Doxomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day3_Palbomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day10_Palbomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day14_Palbomerged_scaleFactornorm_cutsCoverage.bw
/lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day21_Palbomerged_scaleFactornorm_cutsCoverage.bw /lustre/fs4/risc_lab/scratch/jyeung/LS_PDvsDoxo_2/ATACseq/Tn5_offset_merged_bigwigs/scaleFactornorm_cutsCoverage_bigwigs/day28_Palbomerged_scaleFactornorm_cutsCoverage.bw"

samplenames="Cycling day3_Doxo day10_Doxo day14_Doxo day21_Doxo day3_Palbo day10Palbo day14_Palbo day21_Palbo day28_Palbo"

cd /lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/deepTools_files/output_plots
sbatch -p risc,hpc --array=1-10 \
--export=bigwigs="$mergedcuts",\
regions=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/deepTools_files/input_beds/regions.txt,\
samplenames="$samplenames",\
profile_outdir=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/deepTools_files/output_plots/plotProfile,\
heatmap_outdir=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/deepTools_files/output_plots/plotHeatmap,\
filename=/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/deepTools_files/input_beds/filename.txt \
/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/deepTools_files/deeptools_Coverage_1kb.slurm
```

# *FIGURES*

# RNA-seq: save GSEA running score for day 14 Doxo vs Cycling's Hallmark p53 Pathway
```{r}
library(enrichplot)
library(patchwork)
pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/GSEA_day14Doxo_vs_Cyc_Hallmark_p53_pathway.pdf", width=3, height=2.5)
GSEA_plots_H$`Doxo14 vs Cycling0`$`P53 PATHWAY` +theme_classic()
dev.off()

p1 <- gseaplot(GSEA_Results_H$`Doxo14 vs Cycling0`, geneSetID = "P53 PATHWAY", by="runningScore", title = "day 14 Doxo vs Cycling: P53 PATHWAY", color.line="#B3C543", color="#B3C543")+theme_classic()+ylim(-0.4, 0.7)

p2 <- gseaplot(GSEA_Results_H$`Palbo14 vs Cycling0`, geneSetID = "P53 PATHWAY", by="runningScore", title = "day 14 Palbo vs Cycling: P53 PATHWAY", color="#E0ADCD", color.line="#E0ADCD")+theme_classic()+ylim(-0.4, 0.7)

pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/GSEA_day14Doxo_day14Palbo_vs_Cyc_Hallmark_p53_pathway.pdf", width=3, height=5)
plot_grid(p1, p2, ncol=1)
dev.off()
```

# plot heatmap, clustered based on log2FC from day 3 treatment controls & Cycling controls
```{r}

pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/RNA_log2FC_p53_Hallmark_mat.pdf", width=5, height=3.5)
pheatmap(p53_sig_LFCMat, show_rownames = T, show_colnames=F, cluster_cols = F, cluster_rows=F, colorRampPalette(rev(brewer.pal(n = 7, name ="RdBu")))(100), breaks = seq(-6, 6,length.out = 100), annotation_row = clusterDF[ ,c("Clusters", "SASP", "Cell_Cycle", "NFKB_All")], annotation_col = anno_sigLFC, annotation_colors = hm_cols_RNA, border_color = NA, main="all up p53 target genes")

# plot showing only p53 target genes under p53 Hallmark term from GO enrichment. 

p53_genes <- sort(row.names(p53_sig_LFCMat[row.names(p53_sig_LFCMat) %in% unlist(str_split(GO_enrichment_clusters_HallMark_df$Doxo1[GO_enrichment_clusters_HallMark_df$Doxo1$ID %in% "P53 PATHWAY", ]$geneID, "/")), ])) # filter for p53 pathway Hallmark genes that are significantly changing

pheatmap(p53_sig_LFCMat[match(p53_genes, row.names(p53_sig_LFCMat)) , ], show_rownames = T, show_colnames=F, cluster_cols = F, cluster_rows=T, colorRampPalette(rev(brewer.pal(n = 7, name ="RdBu")))(100), breaks = seq(-6, 6,length.out = 100), annotation_row = clusterDF[ ,c("NFKB_All", "SASP")], annotation_col = anno_sigLFC, annotation_colors = hm_cols_RNA, border_color = NA, main="Hallmark p53 pathway: Doxo enriched up genes") # with log2FC legend limits set to -6, 6. 

pheatmap(p53_sig_LFCMat[match(p53_genes, row.names(p53_sig_LFCMat)), ], show_rownames = T, show_colnames=F, cluster_cols = F, cluster_rows=F, colorRampPalette(rev(brewer.pal(n = 7, name ="RdBu")))(100), breaks = seq(-4, 4,length.out = 100), annotation_row = clusterDF[ ,c("NFKB_All", "SASP")], annotation_col = anno_sigLFC, annotation_colors = hm_cols_RNA, border_color = NA, main="Hallmark p53 pathway: Doxo enriched up genes") # with log2FC legend limits set to -4, 4 & gene ids ordered in alphabetical order
dev.off()
```
# plot: venn diagram showing overlap of Doxo or Palbo enriched up genes that are p53 targets, NF-kB targets and SASP. 
```{r}
# write function to create venn diagrams that are proportional to size of each category. 
create_proportional_venn <- function(sets, set_labels, colors = c("#E0ADCD", "#B3C543", "#A0ADCD")) {
  library(eulerr)
  # Check if sets and labels are provided correctly
  if (length(sets) != length(set_labels)) {
    stop("The number of sets and set_labels must be the same.")
  }
  
  # Prepare the data for eulerr
  venn_data <- sets
  names(venn_data) <- set_labels
  
  # Use eulerr to calculate the Venn diagram
  venn_fit <- euler(venn_data)
  
  # Plot the Venn diagram
  plot(venn_fit, 
       fills = colors, 
       labels = list(cex = 1, fontfamily = "Helvetica"),
       quantities = list(cex = 1),
       edges = list(lwd = 2))
}
venn_data_Doxo <- list(row.names(clusterDF[clusterDF$Clusters %in% c("4", "6", "3") & clusterDF$p53_All %in% "Yes", ]), 
                  row.names(clusterDF[clusterDF$Clusters %in% c("4", "6", "3") & clusterDF$NFKB_All %in% "Yes", ]), 
                  row.names(clusterDF[clusterDF$Clusters %in% c("4", "6", "3") & clusterDF$SASP%in% "Yes", ]), 
                  row.names(clusterDF[clusterDF$Clusters %in% c("4", "6", "3"), ]))

venn_data_Palbo <- list(row.names(clusterDF[clusterDF$Clusters %in% c("1", "7") & clusterDF$p53_All %in% "Yes", ]), 
                  row.names(clusterDF[clusterDF$Clusters %in% c("1", "7") & clusterDF$NFKB_All %in% "Yes", ]), 
                  row.names(clusterDF[clusterDF$Clusters %in% c("1", "7") & clusterDF$SASP%in% "Yes", ]), 
                  row.names(clusterDF[clusterDF$Clusters %in% c("1", "7"), ]))

pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/RNA_p53vsNFKB_genes_vennDiagram.pdf", width=4, height=4)

create_proportional_venn(sets=venn_data_Doxo, 
                         set_labels = c("p53 targets", "NF-kB targets", "SASP", "Doxo enriched up genes"), 
                         colors = c("#00B0F0","red", "dark green", "#B3C543")
)

create_proportional_venn(sets=venn_data_Palbo, 
                         set_labels = c("p53 targets", "NF-kB targets", "SASP", "Palbo enriched up genes"), 
                         colors = c("#00B0F0","red", "dark green", "#E0ADCD")
)
dev.off()
```

# plot log2FC of p53 target genes at across all timepoints relative to Cycling. 
```{r}
p53_NFKB_SASP_genes <- clusterDF[clusterDF$p53_All %in% "Yes"|clusterDF$SASP %in% "Yes"|clusterDF$NFKB_All %in% "Yes", ]
p53_NFKB_SASP_genes <- row.names(p53_NFKB_SASP_genes[!p53_NFKB_SASP_genes$Clusters %in% c("5", "2"), ])

log2FC_p53_NFKB_SASP <- rbind(
          data.frame(genes= p53_NFKB_SASP_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% p53_NFKB_SASP_genes, "Palbo3 vs Cycling0"], 
           Condition="day 3 Palbo", 
           Comparison="vs Cycling"),
  
          data.frame(genes= p53_NFKB_SASP_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% p53_NFKB_SASP_genes, "Palbo14 vs Cycling0"], 
           Condition="day 14 Palbo", 
           Comparison="vs Cycling" ), 
           
           data.frame(genes= p53_NFKB_SASP_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% p53_NFKB_SASP_genes, "Palbo21 vs Cycling0"], 
           Condition="day 21 Palbo", 
           Comparison="vs Cycling" ), 
           
           data.frame(genes= p53_NFKB_SASP_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% p53_NFKB_SASP_genes, "Palbo28 vs Cycling0"], 
           Condition="day 28 Palbo", 
           Comparison="vs Cycling" ),
           
           data.frame(genes= p53_NFKB_SASP_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% p53_NFKB_SASP_genes, "Doxo3 vs Cycling0"], 
           Condition="day 3 Doxo", 
           Comparison="vs Cycling"),
      
      data.frame(genes= p53_NFKB_SASP_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% p53_NFKB_SASP_genes, "Doxo14 vs Cycling0"], 
           Condition="day 14 Doxo", 
           Comparison="vs Cycling"), 
      
          data.frame(genes= p53_NFKB_SASP_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% p53_NFKB_SASP_genes, "Doxo21 vs Cycling0"], 
           Condition="day 21 Doxo", 
           Comparison="vs Cycling"), 
      
      data.frame(genes= p53_NFKB_SASP_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% p53_NFKB_SASP_genes, "Palbo14 vs Palbo3"], 
           Condition="day 14 Palbo",
           Comparison="vs day 3"), 
      
       data.frame(genes= p53_NFKB_SASP_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% p53_NFKB_SASP_genes, "Palbo21 vs Palbo3"], 
           Condition="day 21 Palbo",
           Comparison="vs day 3" ), 
           
           data.frame(genes= p53_NFKB_SASP_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% p53_NFKB_SASP_genes, "Palbo28 vs Palbo3"], 
           Condition="day 28 Palbo",
           Comparison="vs day 3" ),
      
      data.frame(genes= p53_NFKB_SASP_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% p53_NFKB_SASP_genes, "Doxo14 vs Doxo3"], 
           Condition="day 14 Doxo",
           Comparison="vs day 3"),
      
      data.frame(genes= p53_NFKB_SASP_genes , 
           log2FC=sig_LFCMat[row.names(sig_LFCMat) %in% p53_NFKB_SASP_genes, "Doxo21 vs Doxo3"], 
           Condition="day 21 Doxo",
           Comparison="vs day 3")
)

log2FC_p53_NFKB_SASP[ ,5:7] <- clusterDF[match(log2FC_p53_NFKB_SASP$genes, row.names(clusterDF)), c("SASP", "NFKB_All", "p53_All")]

# function to create violin plot of log2FC values of genes
create_violin_plot <- function(data=log2FC_p53_NFKB_SASP, plot_title="RNA: log2FC of p53 target genes") {
  library(ggplot2)
  library(ggbeeswarm)
  
  ggplot(
    data,
    aes(x = Condition, y = log2FC, fill = Condition)
  ) +
    geom_violin() +
    geom_quasirandom(
      color = "black", alpha = 0.2,
      method = "pseudorandom", # Aligns jitter points to violin shape
      groupOnX = TRUE,
      size = 0.5
    ) +
    stat_summary(
      fun = "median",
      geom = "crossbar",
      color = "white",
      width = 0.5
    ) +
    theme_classic() +
    theme(
      axis.text.x = element_text(angle = 90, size = 8),
      axis.text.y = element_text(size = 8),
      axis.title.y = element_text(size = 10)
    ) +
    facet_wrap(~Comparison, scales = "free_x") +
    scale_fill_manual(
      values = c(
        "day 3 Doxo" = "#B3C543",
        "day 14 Doxo" = "#B3C543",
        "day 21 Doxo" = "#B3C543",
        "day 3 Palbo" = "#E0ADCD",
        "day 14 Palbo" = "#E0ADCD",
        "day 21 Palbo" = "#E0ADCD",
        "day 28 Palbo" = "#E0ADCD"
      )
    ) +
    ylab("log2FC") +
    xlab("") +
    ggtitle(plot_title)
}

pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/RNA_log2FC_p53_targets.pdf", width=7, height=3)
library(ggbeeswarm)
create_violin_plot(
  data = log2FC_p53_NFKB_SASP[log2FC_p53_NFKB_SASP$p53_All %in% "Yes", ],
  plot_title = "RNA: log2FC of p53 target genes"
)
dev.off()

pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/RNA_day3_log2FC_p53_targets.pdf", width=4, height=3)
library(ggbeeswarm)
create_violin_plot(
  data = log2FC_p53_NFKB_SASP[log2FC_p53_NFKB_SASP$p53_All %in% "Yes" & log2FC_p53_NFKB_SASP$Condition %in% c("day 3 Doxo", "day 3 Palbo"), ],
  plot_title = "RNA: log2FC of p53 target genes"
)
dev.off()

# plot only for day 14
pdf("/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/RNA_day14_log2FC_p53_targets.pdf", width=4, height=3)
library(ggbeeswarm)
create_violin_plot(
  data = log2FC_p53_NFKB_SASP[log2FC_p53_NFKB_SASP$p53_All %in% "Yes" & log2FC_p53_NFKB_SASP$Condition %in% c("day 14 Doxo", "day 14 Palbo"), ],
  plot_title = "RNA: log2FC of p53 target genes"
)
dev.off()

# get significance of log2FC values between day 14 Palbo & day 14 Doxo vs Cycling based on paired student t-test

# Check normality of differences 
d14_Palbo_log2FC <- log2FC_p53_NFKB_SASP[log2FC_p53_NFKB_SASP$p53_All %in% "Yes" & log2FC_p53_NFKB_SASP$Condition %in% "day 14 Palbo" & log2FC_p53_NFKB_SASP$Comparison %in% "vs Cycling", ]$log2FC # log2FC values of p53 targets in day 14 Palbo vs Cycling

d14_Doxo_log2FC <- log2FC_p53_NFKB_SASP[log2FC_p53_NFKB_SASP$p53_All %in% "Yes" & log2FC_p53_NFKB_SASP$Condition %in% "day 14 Doxo" & log2FC_p53_NFKB_SASP$Comparison %in% "vs Cycling", ]$log2FC # log2FC values of p53 targets in day 14 Doxo vs Cycling

differences <- d14_Palbo_log2FC - d14_Doxo_log2FC

# Shapiro-Wilk test for normality
shapiro_test <- shapiro.test(differences)
print(shapiro_test)

# Visualize distribution of differences
ggplot(data.frame(differences), aes(x = differences)) +
  geom_histogram(binwidth = 0.2, fill = "skyblue", color = "black", alpha = 0.7) +
  labs(title = "Distribution of Log2FC Differences", x = "Difference (Drug1 - Drug2)", y = "Count")

# Perform Paired t-test if normality assumption holds
if (shapiro_test$p.value > 0.05) {
  t_test_result <- t.test(d14_Palbo_log2FC, d14_Doxo_log2FC, paired = TRUE)
  print(t_test_result)
} else {
  # Use Wilcoxon signed-rank test if normality is violated
  wilcox_test_result <- wilcox.test(d14_Palbo_log2FC, d14_Doxo_log2FC, paired = TRUE)
  print(wilcox_test_result)
}

```

# RNA-seq Plot normalized counts of well-known p53 targets, MDM2 & CDKN1A over time. 
```{r}
NormCounts_df$Timepoint <- as.numeric(as.character(NormCounts_df$Timepoint))
# save plot to pdf
pdf(file="/lustre/fs4/risc_lab/scratch/jyeung/Manuscript_Figures/Figure_p53_activity_is_different_depending_on_the_presence_of_DNA_damage/RNA_NormCountsMat_MDM2_CDKN1A.pdf", width=8, height=2)

plot_grid(ggplot(NormCounts_df[NormCounts_df$Transcript %in% "CDKN1A", ], aes(x=Timepoint, y=Counts, color=Treatment, shape=Biorep))+geom_point(size=3)+scale_color_manual(values=hm_cols_RNA$Treatment)+theme_minimal()+ylab("Normalized Counts")+xlab("Length of Treatment (days)")+ggtitle("CDKN1A")+scale_y_continuous(n.breaks=5)+scale_x_continuous(n.breaks=5)+theme_classic()+theme(
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8),
      axis.title.y = element_text(size = 10), 
      axis.title.x = element_text(size = 10)
    ),

ggplot(NormCounts_df[NormCounts_df$Transcript %in% "MDM2", ], aes(x=Timepoint, y=Counts, color=Treatment, shape=Biorep))+geom_point(size=3)+scale_color_manual(values=hm_cols_RNA$Treatment)+theme_minimal()+ylab("Normalized Counts")+xlab("Length of Treatment (days)")+ggtitle("MDM2")+scale_y_continuous(n.breaks=5)+scale_x_continuous(n.breaks=5)+theme_classic()+theme(
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8),
      axis.title.y = element_text(size = 10), 
      axis.title.x = element_text(size = 10)
    )+ylim(0,150000)
)

dev.off()
```

